<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
  <head>
    <title>Statistical Methods for  High Dimensional Biology STAT/BIOF/GSAT 540</title>
    <meta charset="utf-8" />
    <link href="libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link href="libs/remark-css-0.0.1/default-fonts.css" rel="stylesheet" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Statistical Methods for <br>High Dimensional Biology<br><font size=8>STAT/BIOF/GSAT 540</font>
## <font color=red> Lecture 6 – ANOVA and Linear Models</font> <br style=line-height:2> Gabriela Cohen Freue
### <font size=5> Other contributors: Jenny Bryan, Sara Mostafavi, Su-In Lee</font>

---

class: middle, center
### &lt;font color="red"&gt; Are these genes truly different in NrlKO compared to WT?&lt;/font&gt;

### H&lt;sub&gt;0&lt;/sub&gt;: the expression level of gene A is the same in both conditions.

### Is there **enough** evidence in the data to reject H&lt;sub&gt;0&lt;/sub&gt;?


![](L6_ANOVA_files/figure-html/unnamed-chunk-2-1.svg)&lt;!-- --&gt;

---
class: middle
### **Statistics**: use a random sample to learn about the population
![](L6_ANOVA_files/slide3.png)
---
class: middle

# Last class: hypothesis testing

### 1. Define a &lt;font color = "red"&gt;test statistic&lt;/font&gt; to test `\(H_0\)`
- 2-sample *t*-test
- Welch *t*-test
- Wilcoxon rank-sum test
- Kolmogorov-Smirnov test

### 2. Compute the &lt;font color=red&gt;observed value&lt;/font&gt; for the test statistic
### 3. Compute the probability of seeing a test statistic as extreme as that observed, under the &lt;font color = "red"&gt;null sampling distribution&lt;/font&gt; (p-value) 
### 4. Make a decision about the &lt;font color = "red"&gt;significance&lt;/font&gt; of the results, based on a pre-specified value (alpha, significance level)
---
class: middle
## We can run these tests in R. 
### Example: use the `t.test` function to test `\(H_0\)` using a classical 2-sample *t*-test.


```r
miniDat %&gt;% subset(gene=="Irs4")%&gt;% t.test(gExp ~ gType, data=., var.equal = TRUE)
```

```
## 
## 	Two Sample t-test
## 
## data:  gExp by gType
## t = -0.52865, df = 37, p-value = 0.6002
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.12597002  0.07383844
## sample estimates:
## mean in group NrlKO    mean in group wt 
##            7.739684            7.765750
```
---
class: middle
# Today...

### - review other methods to test statistical hypothesis

### - extend tests to more than 2-groups comparisons

### - methods to analyze the association of multiple variables with the response
---
class: middle

![](L6_ANOVA_files/slide4.png)
---
class: middle
It seems that we can use any of these methods to test `\(H_0\)`

![](L6_ANOVA_files/slide5.png)
---
class: middle
## *t*-test *vs* linear regression: &lt;font color = "red"&gt;why the same results?&lt;/font&gt;


```r
irs4Dat &lt;- subset(miniDat,gene=="Irs4")
ttest.irs4&lt;-t.test(gExp ~ gType, irs4Dat, var.equal = TRUE)
list("t value"=ttest.irs4$stat,"p-value"=ttest.irs4$p.value)
```

```
## $`t value`
##          t 
## -0.5286494 
## 
## $`p-value`
## [1] 0.6002058
```



```r
lm.irs4 &lt;- summary(lm(gExp ~ gType, irs4Dat))
list("t value"=lm.irs4$coeff[2,3],"p-value"=lm.irs4$coeff[2,4])
```

```
## $`t value`
## [1] 0.5286494
## 
## $`p-value`
## [1] 0.6002058
```

---
class: middle
## *t*-test *vs* linear regression: &lt;font color = "red"&gt;where's the line?&lt;/font&gt;

![](L6_ANOVA_files/figure-html/unnamed-chunk-6-1.svg)&lt;!-- --&gt;

---
class: middle
# From *t*-test to linear regression
Let's change the notation to give a common framework to all methods
&lt;br&gt; &lt;br&gt;
&lt;font size=5&gt; `$$Y \sim G; \; E[Y] = \mu_Y$$` &lt;/font&gt;
 **&lt;center&gt;↓&lt;/center&gt;**

&lt;font size=5&gt; `$$Y = \mu_Y + \varepsilon_Y; \; \varepsilon_Y \sim G; \; E[\varepsilon_Y] = 0$$` &lt;/font&gt;
&lt;br&gt;
We can use a subindeces to distinguish observations from each group, i.e., 

&lt;font size=5&gt; `$$Y_{ij} = \mu_j + \varepsilon_{ij};\; \; \varepsilon_{ij} \sim G_j; \; \;E[\varepsilon_{ij}] = 0;$$` &lt;/font&gt;
&lt;br&gt;
where `\(j = \textrm{\{wt, NrlKO}\}\)` or `\(j=\textrm{\{1, 2}\}\)`; and `\(i=1, \ldots, n_j\)`.
&lt;br&gt;

 &gt; For example: `\(Y_{11}\)` is the first observation in group 1 or wt
 
---
class: middle

## The goal is to test 

### `$$H_0 : \mu_1 = \mu_2$$`

### using data from the model

&lt;font size=5&gt; `$$Y_{ij} = \mu_j + \varepsilon_{ij};\; \; \varepsilon_{ij} \sim G; \; \;E[\varepsilon_{ij}] = 0;$$`&lt;/font&gt;
&lt;br&gt;
where `\(j = \textrm{\{wt, NrlKO}\}\)` or `\(j=\textrm{\{1, 2}\}\)`; and `\(i=1, \ldots, n_j\)`.
&lt;br&gt;

&gt; For simplicity, we assume a common distribution `\(G\)` for all groups

### the model is written with a &lt;font color = "red"&gt;cell-means&lt;/font&gt; - `\(\mu_j\)` - parametrization
---
class: middle

### Note that for each group, the population mean is given by 
### `$$E[Y_{ij}] = \mu_j,$$` 

###A natural **estimator** of the population mean is the &lt;font color = "red"&gt;**sample mean**&lt;/font&gt;

### Classical methods for *t*-test, ANOVA, and linear regression use the group sample means as estimators.

See, for example, the `t.test` function in R:


```r
ttest.irs4$estimate
```

```
## mean in group NrlKO    mean in group wt 
##            7.739684            7.765750
```
---
class: middle
### However, the `lm` function reports other estimates, &lt;font color = "red"&gt;why?&lt;/font&gt;  


```r
(means.irs4&lt;-as.data.frame(irs4Dat %&gt;% group_by(gType) %&gt;% summarize(meanGroups=mean(gExp,digits=6))))
```

```
##   gType meanGroups
## 1 NrlKO   7.739684
## 2    wt   7.765750
```


```r
lm.irs4$coefficients[,1]
```

```
## (Intercept)     gTypewt 
##  7.73968421  0.02606579
```
`\(\hspace{3em}\)` **↓**
&lt;font color = "red"&gt;sample mean of NrlKO group&lt;/font&gt; 


### But the second coefficient is &lt;font color = "red"&gt;**not**&lt;/font&gt; the sample mean of the WT group
---
class: middle

### By default, the `lm` function uses a different &lt;font color=red&gt;parametrization&lt;/font&gt; since the goal is to *compare* the means, not to study each in isolation

&lt;font size=5&gt; From &lt;font color=red&gt;**cell-means**&lt;/font&gt; - `\(\mu_j\)`:  `$$Y_{ij} = \mu_j + \varepsilon_{ij};\; \; \varepsilon_{ij} \sim G; \; \;E[\varepsilon_{ij}] = 0;$$`
**&lt;center&gt;↓&lt;/center&gt;**
to &lt;font color = "red"&gt;**reference-treatment effect**&lt;/font&gt; - `\((\theta,\tau_j)\)`: `$$Y_{ij} = \theta+\tau_j + \varepsilon_{ij};\; \; \tau_1=0, \; \; \varepsilon_{ij} \sim G; \; \;E[\varepsilon_{ij}] = 0;$$`

### Note that for each group, the population mean is given by 
&lt;font size=5&gt; `$$E[Y_{ij}] = \theta+\tau_j=\mu_j, \; \text{and}$$`
`$$\tau_2=\mu_2-\mu_1$$` 
---
class: middle
## Relation between parametrizations

![](L6_ANOVA_files/slide16.png)

---
class: middle

### `lm` reports the sample mean of the &lt;font color = "red"&gt;reference&lt;/font&gt; group (WT): `\(\hat\theta\)`

### and the &lt;font color = "red"&gt;treatment effect&lt;/font&gt;, i.e., difference between the sample means of both groups: `\(\hat\tau_2\)`
 

```r
lm.irs4$coefficients[,1]
```

```
## (Intercept)     gTypewt 
##  7.73968421  0.02606579
```

```r
data.frame(meanWT=means.irs4[1,2],meanDiff=diff(means.irs4$meanGroups))
```

```
##     meanWT   meanDiff
## 1 7.739684 0.02606579
```
---
class: middle
### The &lt;font color = "red"&gt;reference-treatment effect &lt;/font&gt; parametrization is also used by other methods

### Note that the 2-sample *t*-test presented last class equals

&lt;font size=5&gt;$$t=\frac{\hat \tau_2}{\sqrt{V(\hat \tau_2})},$$&lt;/font&gt;

&gt; `\(\hat \tau_2=\bar{Y_2}-\bar{Y_1}\)` is the difference of sample means




```r
diff(means.irs4$meanGroups)/sqrt(s2Diff.irs)
```

```
##      Irs4 
## 0.5288651
```

```r
ttest.irs4$statistic
```

```
##          t 
## -0.5286494
```
---
class: middle

### Similarly, `aov` uses `\(\hat \tau_2\)` to build the *F*-statistic



```r
summary(aov(gExp ~gType, irs4Dat))
```

```
##             Df Sum Sq Mean Sq F value Pr(&gt;F)
## gType        1 0.0066 0.00662   0.279    0.6
## Residuals   37 0.8764 0.02369
```
### It can be proved mathematically that 
### `$$F=t^2$$` 
Here numerically,


```r
as.numeric(ttest.irs4$statistic^2)
```

```
## [1] 0.2794701
```
---
class: middle
### What if there are more than 2 groups?
![](L6_ANOVA_files/slide17.png)
---
class: middle

# Beyond 2-groups comparisons


## &lt;font color = "red"&gt;t-test&lt;/font&gt; 
&gt; Special case of &lt;font color = "red"&gt;ANOVA&lt;/font&gt;, but with ANOVA you can compare **more than two groups** and **more than one factor**.

&lt;br&gt;

## &lt;font color = "red"&gt;ANOVA&lt;/font&gt; 
&gt; Special case of &lt;font color = "red"&gt;linear regression&lt;/font&gt;, but with linear regression you can model **many quantitative and qualitative variables**. 

---
class: middle

# But where's the line??
![](L6_ANOVA_files/figure-html/unnamed-chunk-15-1.png)&lt;!-- --&gt;
---
class: middle
### Using a special matrix `\(X\)`, we can write our model in matrix notation:

![](L6_ANOVA_files/model_matrix.png)
### and "a line" seems to appear!!
---
class: middle

&lt;img src="L6_ANOVA_files/model_matrix_s.png" height="350px" /&gt;

&gt; &lt;font color = "red"&gt;Note that `\(Y_{i1}= 1 \times \theta + 0 \times \tau_2 + 0 \times \tau_3 + \varepsilon_{i1}=\theta + \varepsilon_{i1}\)`

&gt; &lt;font color = "blue"&gt;Note that `\(Y_{i2}= 1 \times \theta + 1 \times \tau_2 + 0 \times \tau_3 + \varepsilon_{i2}=\theta + \tau_2+\varepsilon_{i2}\)`

&gt; &lt;font color = "green"&gt;Note that `\(Y_{i3}= 1 \times \theta + 0 \times \tau_2 + 1 \times \tau_3 + \varepsilon_{i3}=\theta + \tau_3+\varepsilon_{i3}\)`

&lt;br&gt;

&lt;font size=5 color="black"&gt; 
Which is the same as `\(\; \rightarrow \; Y_{ij} = \theta + \tau_j + \varepsilon_{ij}, \; \tau_1=0\)`
---
class: middle
### Difference of means: the "treatment effect"
&lt;img src="L6_ANOVA_files/slide18.png" height="320px" /&gt;

&gt; Note that `\(E[Y_{ij}] = \theta + \tau_j + E[\varepsilon_{ij}]=\theta +\tau_{j}\)`

`\(E[Y_{i1}]=\theta\)`
&lt;br&gt;

`\(E[Y_{i2}]=\theta+\tau_2 \; \rightarrow \tau_2=E[Y_{i2}]-E[Y_{i1}]=\mu_2-\mu_1\)`
&lt;br&gt;

`\(E[Y_{i3}]=\theta+\tau_3 \; \rightarrow \tau_3=E[Y_{i3}]-E[Y_{i1}]=\mu_3-\mu_1\)`

---
class: middle
![](L6_ANOVA_files/linear_form.png)
---
class: middle

### &lt;font color = "red"&gt;Linear regression&lt;/font&gt; can include **quantitative &amp; qualitative covariates**. 

&lt;center&gt;
&lt;img src="L6_ANOVA_files/LM_vbles.png" height="470px" /&gt;
&gt; Linear in the parameters `\(\alpha\)`: `\(X\)` can contain `\(x^2\)`, `\(log(x)\)`, etc.

---
class: middle, center

## How it works in practice using &lt;font color=red&gt;lm()&lt;/font&gt; in R

&lt;big&gt;

## `$$Y = X\alpha + \varepsilon$$` 
**↓**
&lt;br&gt;

```
lm(y ~ x, data = yourData)
```
&lt;/font&gt;
.pull-left[
&lt;font color = red&gt; y ~ x: &lt;/font&gt; formula, y numeric, x numeric and/or factor
]

.pull-right[
&lt;font color = red&gt; yourData: &lt;/font&gt; data.frame in which x and y are to be found (we recommend this style)
]
&lt;br&gt;
### By default, R uses a ref-tx parametrization but you can control that!
---
---
class: middle
## Beyond 2-group comparisons in our case study:

### &lt;font color="red"&gt; Is the expression of gene A the same at all developmental stages?&lt;/font&gt;

`$$H_0 : \mu_{E16} = \mu_{P2} = \mu_{P6} = \mu_{P10} = \mu_{4weeks}$$`
&lt;center&gt;


![](L6_ANOVA_files/figure-html/unnamed-chunk-20-1.svg)&lt;!-- --&gt;
---
class: middle

```r
with(miniDat,
     tapply(gExp, list(devStage, gene), mean))
```

```
##          theBore   theHit
## E16     7.544143 5.540857
## P2      7.583500 5.844875
## P6      7.554000 5.784250
## P10     7.571000 6.375125
## 4_weeks 7.559000 9.173375
```
&lt;center&gt;
![](L6_ANOVA_files/figure-html/unnamed-chunk-22-1.svg)&lt;!-- --&gt;

---
class: middle

```r
means.dev &lt;- as.data.frame(miniDat %&gt;% group_by(devStage) %&gt;% summarize(cellMeans=mean(gExp)))
means.dev
```

```
##   devStage cellMeans
## 1      E16  6.542500
## 2       P2  6.714187
## 3       P6  6.669125
## 4      P10  6.973063
## 5  4_weeks  8.366188
```

```r
data.frame(cellMeans=means.dev[1,2],txEffects=diff(means.dev$cellMeans))
```

```
##   cellMeans  txEffects
## 1    6.5425  0.1716875
## 2    6.5425 -0.0450625
## 3    6.5425  0.3039375
## 4    6.5425  1.3931250
```
![](L6_ANOVA_files/slide25.png)
---
class: middle
![](L6_ANOVA_files/figure-html/unnamed-chunk-24-1.svg)&lt;!-- --&gt;
![](L6_ANOVA_files/slide26.png)
---
class: middle
![](L6_ANOVA_files/figure-html/unnamed-chunk-25-1.svg)&lt;!-- --&gt;
![](L6_ANOVA_files/slide27.png)
---
class: middle
![](L6_ANOVA_files/figure-html/unnamed-chunk-26-1.svg)&lt;!-- --&gt;
![](L6_ANOVA_files/slide28.png)
---
class: middle
&lt;big&gt;
&lt;font size = 5&gt;$$Y = X \alpha + \varepsilon$$&lt;/font&gt;
`$$\alpha = (\theta, \tau_{P2}, \tau_{P6}, \tau_{P10}, \tau_{4weeks})$$`

### in the context of this model, we generally test null hypothesis of two types:
.pull-left[
&lt;center&gt;
`$$H_0: \tau_j = 0$$`
vs
`$$H_0: \tau_j \neq 0$$`
for each *j* individually
]

.pull-right[
&lt;center&gt;
`$$H_0: \tau_j = 0$$`
vs
`$$H_0: \tau_j \neq 0$$`
for all *j* at the same time
]
---
class: middle
![](L6_ANOVA_files/slide30.png)
---
class: middle

## F-test and overall significance of one or more covariates
&lt;big&gt;
- the t-stat in linear regression allows us to test simple hypotheses:
`$$H_0 : \tau_i = 0$$`
`$$H_A : \tau_j \neq 0$$`
- But when we have multiple covariates/factors, we often like to test more complex hypotheses: 
`$$H_0 : \tau_2 = \tau_3 = ... = 0\textrm{ [AND statement]}$$`
`$$H_A : \tau_i \neq 0 \textrm{ for some i [OR statement]}$$`
- F-test allows us to test such compound tests
---
class: middle
### Increasing the complexity of our linear regression model ...
What if you have two categorical variables: genotype and time
(we will simplify the example by only considering two time points E16 and 4weeks)

&gt; ### Is the assocation between gene expression and genotype different at different developmental stages?

![](L6_ANOVA_files/slide32.png)
---
class: middle
![](L6_ANOVA_files/slide33.png)
---
class: middle
![](L6_ANOVA_files/slide34.png)
---
class: middle
![](L6_ANOVA_files/slide35.png)
---
class: middle
![](L6_ANOVA_files/slide36.png)
---
class: middle
![](L6_ANOVA_files/slide37.png)
---
class: middle
![](L6_ANOVA_files/slide38.png)
---
class: middle

## Let's go through some example genes to get a sense of what an interaction effect looks like.

We have three parameters we'd like to interpret:
.pull-left[
&gt; ### Main effect: genotype
### Main effect: age
### Interaction: genotype*age
]
.pull-right[
&lt;br&gt;
![](L6_ANOVA_files/slide39.png)
]
---
class: middle
![](L6_ANOVA_files/slide40.png)
---
class: middle
![](L6_ANOVA_files/slide41.png)
---
class: middle
![](L6_ANOVA_files/slide42.png)
---
class: middle
![](L6_ANOVA_files/slide43.png)
---
class: middle
![](L6_ANOVA_files/slide44.png)
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"highlightSpans": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();</script>

<script>
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
