---
title: "Statistical Methods for <br>High Dimensional Biology<br><font size=8>STAT/BIOF/GSAT 540</font>"
subtitle: "<font color=red> Lecture 6 -- ANOVA and Linear Models</font> <br style=line-height:2> Gabriela Cohen Freue"
date: "<font size=5> Other contributors: Jenny Bryan, Sara Mostafavi, Su-In Lee</font>"
output:
  xaringan::moon_reader:
    lib_dir: libs
    nature:
      highlightStyle: github
      highlightLines: true
      highlightSpans: true
      countIncrementalSlides: false
--- 
class: middle, center
### <font color="red"> Are these genes truly different in NrlKO compared to WT?</font>

### H<sub>0</sub>: the expression level of gene A is the same in both conditions.

### Is there **enough** evidence in the data to reject H<sub>0</sub>?
```{r, include=FALSE}
library(dplyr)
library(ggplot2)
library(ggthemes)
library(grid)
library(gridExtra)
library(lattice)
library(latticeExtra)

jCols <- c(x = "blue", y = "orange")
trellis.par.set(superpose.symbol = list(col = jCols),
                superpose.line = list(col = jCols))
jCex <- 3 

prDes <- readRDS("data/GSE4051_design.rds")
str(prDes)

prDat<-read.table("data/GSE4051_data.tsv",
                      sep = "\t", header = T, row.names = 1)
str(prDat, list.len = 10)

miniDat <- as.vector(t(prDat[c("1422248_at", "1450946_at"), ]))
miniDat <- data.frame(gene = rep(c("Irs4", "Nrl"), each = nrow(prDes)),
                      gExp = miniDat)
miniDat <- data.frame(prDes, miniDat) # ignore the warning about row names
miniDat$gType <- factor(miniDat$gType, rev(levels(miniDat$gType)))
str(miniDat)
```

```{r echo=FALSE, fig.height=3.5, dev='svg'}
irsDat <- filter(miniDat, gene == "Irs4")
nrlDat <- filter(miniDat, gene == "Nrl")

irsLim <- ggplot(irsDat, aes(x = gExp, y = gType, colour = gType)) + 
             geom_point(alpha = 0.5) +
             labs(title = "Irs4 Gene Expression") +
             theme_bw() +
             theme(legend.position = "none") +
             xlim(5, 15)

nrlLim <- ggplot(nrlDat, aes(x = gExp, y = gType, colour = gType)) + 
             geom_point(alpha = 0.5) +
             labs(title = "Nrl Gene Expression") +
             theme_bw() +
             theme(legend.position = "none") +
             xlim(5, 15)

options(repr.plot.width=8, repr.plot.height=5)

grid.arrange(irsLim, nrlLim, ncol = 1)
```

---
class: middle
### **Statistics**: use a random sample to learn about the population
![](L6_ANOVA_files/slide3.png)
---
class: middle

# Last class: hypothesis testing

### 1. Define a **test statistic** to test $H_0$
- 2-sample *t*-test
- Welch *t*-test
- Wilcoxon rank-sum test
- Kolmogorov-Smirnov test

### 2. Compute the <font color=red>observed value</font> for the test statistic
### 3. Compute the probability of seeing a test statistic as extreme as that observed, under the **null sampling distribution** (p-value) 
### 4. Make a decision about the significance of the results, based on a pre-specified value (alpha, significance level)
---
class: middle
## We can run these tests in R. 
### Example: use the `t.test` function to test $H_0$ using a classical 2-sample *t*-test.

``` {r}
miniDat %>% subset(gene=="Irs4")%>% t.test(gExp ~ gType, data=., var.equal = TRUE)
```
---
class: middle
# Today...

### - review other methods to test statistical hypothesis

### - extend tests to more than 2-groups comparisons

### - methods to analyze the effect of multiple factors on a response
---
class: middle

![](L6_ANOVA_files/slide4.png)
---
class: middle
It seems that we can use any of these methods to test $H_0$

![](L6_ANOVA_files/slide5.png)
---
class: middle
## *t*-test *vs* linear regression: <font color = "red">why the same results?</font>

``` {r}
irs4Dat <- subset(miniDat,gene=="Irs4")
ttest.irs4<-t.test(gExp ~ gType, irs4Dat, var.equal = TRUE)
list("t value"=ttest.irs4$stat,"p-value"=ttest.irs4$p.value)
```


```{r}
lm.irs4 <- summary(lm(gExp ~ gType, irs4Dat))
list("t value"=lm.irs4$coeff[2,3],"p-value"=lm.irs4$coeff[2,4])
```

---
class: middle
## *t*-test *vs* linear regression: <font color = "red">where's the line?</font>

```{r echo=FALSE, fig.width=5, fig.height= 3, dev='svg'}
irsDat <- filter(miniDat, gene == "Irs4")
nrlDat <- filter(miniDat, gene == "Nrl")

irsLim <- ggplot(irsDat, aes(x = gExp, y = gType, colour = gType)) + 
             geom_point(alpha = 0.5) +
             labs(title = "Irs4 Gene Expression") +
             theme_bw() +           
             theme(legend.position = "none") +
             xlim(5, 15)

nrlLim <- ggplot(nrlDat, aes(x = gExp, y = gType, colour = gType)) + 
             geom_point(alpha = 0.5) +
             labs(title = "Nrl Gene Expression") +
             theme_bw() +
             theme(legend.position = "none") +
             xlim(5, 15)

grid.arrange(irsLim, nrlLim, ncol = 1)
```

---
class: middle
# From *t*-test to linear regression
Let's change the notation to present the problem as a regression
<br> <br>
$$Y \sim G; \; E[Y] = \mu_Y$$
 **<center>↓</center>**

$$Y = \mu_Y + \varepsilon_Y; \; \varepsilon_Y \sim G; \; E[\varepsilon_Y] = 0$$
<br>
We can use a subindeces to distinguish observations from each group, i.e., 

$$Y_{ij} = \mu_j + \varepsilon_{ij};\; \; \varepsilon_{ij} \sim G_j; \; \;E[\varepsilon_{ij}] = 0;$$
<br>
where $j = \textrm{\{wt, NrlKO}\}$ or $j=\textrm{\{1, 2}\}$; and $i=1, \ldots, n_j$.
<br>

 > For example: $Y_{11}$ is the first observation in group 1 or wt
 
---
class: middle

## The goal is to test 

### $$H_0 : \mu_1 = \mu_2$$

### using data from the model

### $$Y_{ij} = \mu_j + \varepsilon_{ij};\; \; \varepsilon_{ij} \sim G; \; \;E[\varepsilon_{ij}] = 0;$$
<br>
where $j = \textrm{\{wt, NrlKO}\}$ or $j=\textrm{\{1, 2}\}$; and $i=1, \ldots, n_j$.
<br>

> For simplicity, we assume a common distribution $G$ for all groups

---
class: middle

### Note that for each group, the population mean is given by 
### $$E[Y_{ij}] = \mu_j,$$ 

###A natural **estimator** of the population mean is the <font color = "red">**sample mean**</font>

### By default, the `t.test` function uses the group sample means as estimators.  

```{r}
ttest.irs4$estimate
```
---
class: middle
### However, the `lm` function reports other estimates, <font color = "red">why?</font>  

```{r}
lm.irs4$coefficients[,1]
```
$\hspace{3em}$ **↓**
<font color = "red">sample mean of NrlKO group</font> 


### But the second coefficient is <font color = "red">**not**</font> the sample mean of the WT group

```{r}
ttest.irs4$estimate
```
---
class: middle
### By default, the `lm` function assumes a different *parametrization*:

### $$Y_{ij} = \mu_j + \varepsilon_{ij};\; \; \varepsilon_{ij} \sim G; \; \;E[\varepsilon_{ij}] = 0;$$
###**<center>↓</center>**
### $$Y_{ij} = \theta+\tau_j + \varepsilon_{ij};\; \; \tau_1=0, \; \; \varepsilon_{ij} \sim G; \; \;E[\varepsilon_{ij}] = 0;$$

### Note that for each group, the population mean is given by 
### $$E[Y_{ij}] = \theta+\tau_j=\mu_j,$$ 

###The first model uses a <font color = "red">*cell-means* parametrization</font>
###The second model uses a <font color = "red">*reference-treatment effect* parametrization</font>
---
class: middle
## Relation between parametrizations

![](L6_ANOVA_files/slide16.png)

---
class: middle
### By default, R uses the <font color = "red">*reference-treatment effect* parametrization</font> 

$$Y_{ij} = \theta+\tau_j + \varepsilon_{ij};\; \; \tau_1=0, \; \; \varepsilon_{ij} \sim G; \; \;E[\varepsilon_{ij}] = 0;$$

This is the sample mean of the *reference* group (wt in this case): $\hat\theta$
```{r}
lm.irs4$coefficients[1,1]
```
 
and this is the *treatment effect*, i.e., the difference between the sample means of both groups: $\hat\tau_2$
 
```{r}
lm.irs4$coefficients[2,1]
means.irs4<-irs4Dat %>% group_by(gType) %>% summarize(meanGroups=mean(gExp,digits=6)) 
diff(means.irs4$meanGroups)
```
---
class: middle
### Beyond 2-groups comparisons:
![](L6_ANOVA_files/slide17.png)
---
class: middle
<big>
<font size = 12> $$Y = X\alpha + \varepsilon$$ </font>
<font color = red> $Y:$ </font> The column vector of the responses, one element per experimental unit

<font color = red> $X:$ </font> a (design) matrix that represents covariate info, one row per experimental unit

<font color = red> $\alpha:$ </font> a column vector of parameters in the linear model

<font color = red> $\varepsilon:$ </font> a column vector of the errors
<br>
### Generic linear model, using conventional matrix formulation

---
class: middle

# Beyond 2-groups comparisons


## <font color = "red">t-test</font> 
> Special case of <font color = "red">ANOVA</font>, but with ANOVA you can compare **more than two groups** and **more than one factor**.

<br>

## <font color = "red">ANOVA</font> 
> Special case of <font color = "red">linear regression</font>, but with linear regression you can model **many quantitative and qualitative variables**. 

---

---
class: middle
![](L6_ANOVA_files/slide18.png)
---
class: middle
![](L6_ANOVA_files/slide19.png)
---
class: middle
![](L6_ANOVA_files/slide20.png)
---
class: middle, center

## How it works in practice using <font color=red>lm()</font> in R

<big>

<font size = 5> $$Y = X\alpha + \varepsilon$$ **↓**
<br>
```
lm(y ~ x, data = jDat)
```
</font>
.pull-left[
<font color = red> y ~ x: </font> formula, y numeric, x factor
]

.pull-right[
<font color = red> jDat: </font> optional data.frame in which x and y are to be found (I recommend this style)
]
<br>
### By default, R uses a ref-tx parametrization but you can control that!(called "contr.treatment"; see?contrasts)
---
class: middle
![](L6_ANOVA_files/slide22.png)
---
class: middle
## Back to our example!
### <font color="red"> Is the gene expression of gene A different at different developmental stages?</font>

$$H_0 : \mu_{E16} = \mu_{P2} = \mu_{P6} = \mu_{P10} = \mu_{4weeks}$$
<center>
```{r, include=FALSE}
library(lattice)

prDes <- readRDS("data/GSE4051_design.rds")

prDat<-read.table("data/GSE4051_data.tsv",
                      sep = "\t", header = T, row.names = 1)

## I've selected this as our hit
theHit <- which(rownames(prDat) == "1440645_at") # 17843
## and this as our boring gene
theBore <- which(rownames(prDat) == "1443184_at") # 18898

keepers <- data.frame(row = c(theBore, theHit),
                       probesetID = I(rownames(prDat)[c(theBore, theHit)]))

miniDat <- as.vector(t(prDat[keepers$probesetID, ]))
miniDat <- data.frame(gene = rep(c("theBore", "theHit"), each = nrow(prDes)),
                      gExp = miniDat)
miniDat <- data.frame(prDes, miniDat)

boreDat <- filter(miniDat, gene == "theBore")
hitDat <- filter(miniDat, gene == "theHit")
```

```{r, echo=FALSE, fig.height= 3, dev='svg'}
boreLim <- ggplot(boreDat, aes(x = devStage, y = gExp)) + 
             geom_jitter(width = 0.2, alpha = 0.5) +
             labs(title = "theBore") +
             theme_bw() +
             theme(legend.position = "none") +
             ylim(5, 10) +
             xlab("") +
             stat_summary(aes(group=1), fun.y=mean, geom="line", colour="red")

hitLim <- ggplot(hitDat, aes(x = devStage, y = gExp)) + 
             geom_jitter(width = 0.2, alpha = 0.5) +
             labs(title = "theHit") +
             theme_bw() +
             theme(legend.position = "none") +
             ylim(5, 10) +
             ylab("") +
             xlab("") +
             stat_summary(aes(group=1), fun.y=mean, geom="line", colour="red")

grid.arrange(boreLim, hitLim, nrow = 1)
```
---
class: middle
``` {r}
with(miniDat,
     tapply(gExp, list(devStage, gene), mean))
```
<center>
```{r, echo=FALSE, fig.height= 3, dev='svg'}
grid.arrange(boreLim, hitLim, nrow = 1)
```
---
class: middle
![](L6_ANOVA_files/slide25.png)
---
class: middle
```{r, echo=FALSE, fig.height= 3, dev='svg'}
hitLim
```
![](L6_ANOVA_files/slide26.png)
---
class: middle
```{r, echo=FALSE, fig.height= 3, dev='svg'}
hitLim
```
![](L6_ANOVA_files/slide27.png)
---
class: middle
```{r, echo=FALSE, fig.height= 3, dev='svg'}
hitLim
```
![](L6_ANOVA_files/slide28.png)
---
class: middle
<big>
<font size = 5>$$Y = X \alpha + \varepsilon$$</font>
$$\alpha = (\theta, \tau_{P2}, \tau_{P6}, \tau_{P10}, \tau_{4weeks})$$

### in the context of this model, we generally test null hypothesis of two types:
.pull-left[
<center>
$$H_0: \tau_j = 0$$
vs
$$H_0: \tau_j \neq 0$$
for each *j* individually
]

.pull-right[
<center>
$$H_0: \tau_j = 0$$
vs
$$H_0: \tau_j \neq 0$$
for all *j* at the same time
]
---
class: middle
![](L6_ANOVA_files/slide30.png)
---
class: middle

## F-test and overall significance of one or more covariates
<big>
- the t-stat in linear regression allows us to test simple hypotheses:
$$H_0 : \tau_i = 0$$
$$H_A : \tau_j \neq 0$$
- But when we have multiple covariates/factors, we often like to test more complex hypotheses: 
$$H_0 : \tau_2 = \tau_3 = ... = 0\textrm{ [AND statement]}$$
$$H_A : \tau_i \neq 0 \textrm{ for some i [OR statement]}$$
- F-test allows us to test such compound tests
---
class: middle
### Increasing the complexity of our linear regression model ...
What if you have two categorical variables: genotype and time
(we will simplify the example by only considering two time points E16 and 4weeks)

> ### Is the assocation between gene expression and genotype different at different developmental stages?

![](L6_ANOVA_files/slide32.png)
---
class: middle
![](L6_ANOVA_files/slide33.png)
---
class: middle
![](L6_ANOVA_files/slide34.png)
---
class: middle
![](L6_ANOVA_files/slide35.png)
---
class: middle
![](L6_ANOVA_files/slide36.png)
---
class: middle
![](L6_ANOVA_files/slide37.png)
---
class: middle
![](L6_ANOVA_files/slide38.png)
---
class: middle

## Let's go through some example genes to get a sense of what an interaction effect looks like.

We have three parameters we'd like to interpret:
.pull-left[
> ### Main effect: genotype
### Main effect: age
### Interaction: genotype*age
]
.pull-right[
<br>
![](L6_ANOVA_files/slide39.png)
]
---
class: middle
![](L6_ANOVA_files/slide40.png)
---
class: middle
![](L6_ANOVA_files/slide41.png)
---
class: middle
![](L6_ANOVA_files/slide42.png)
---
class: middle
![](L6_ANOVA_files/slide43.png)
---
class: middle
![](L6_ANOVA_files/slide44.png)