{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# Statistical Methods for High Dimensional Biology (STAT/BIOF/GSAT 540)\n",
    "\n",
    "\n",
    "# <font color=red> Lecture 6: ANOVA and Linear Models"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "subslide"
    }
   },
   "source": [
    "# Overview\n",
    "In this lecture, we will go beyond 2-groups comparisons and use ANOVA to compare these groups.\n",
    "\n",
    "### Table of Contents:\n",
    "\n",
    "1. [**Data description**](#Data-description)\n",
    "    1. [Loading the data](#Loading-the-data)\n",
    "    2. [Subsetting the data](#Subsetting-the-data)\n",
    "\n",
    "### R Dependencies:\n",
    "\n",
    "- knitr\n",
    "- lattice\n",
    "- latticeExtra\n",
    "- RColorBrewer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "slideshow": {
     "slide_type": "skip"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: 'dplyr'\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "Registered S3 methods overwritten by 'ggplot2':\n",
      "  method         from \n",
      "  [.quosures     rlang\n",
      "  c.quosures     rlang\n",
      "  print.quosures rlang\n",
      "\n",
      "Attaching package: 'gridExtra'\n",
      "\n",
      "The following object is masked from 'package:dplyr':\n",
      "\n",
      "    combine\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(knitr)\n",
    "knitr::opts_knit$set(root.dir=\".\")\n",
    "\n",
    "library(dplyr)\n",
    "\n",
    "library(ggplot2)\n",
    "library(ggthemes)\n",
    "\n",
    "library(grid)\n",
    "library(gridExtra)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# Data description\n",
    "\n",
    "Dataset was obtained from: [GSE4051](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE4051)\n",
    "\n",
    "> ## Targeting of GFP to new-born rods by Nrl promoter and temporal expression profiling of flow-sorted photoreceptors\n",
    "Akimoto M, Cheng H, Zhu D, Brzezinski JA, Khanna R, Filippova E, Oh EC, Jing Y, Linares JL, Brooks M, Zareparsi S, Mears AJ, Hero A, Glaser T, Swaroop A.\n",
    "<br><br>\n",
    "The Maf-family transcription factor Nrl is a key regulator of photoreceptor differentiation in mammals. Ablation of the Nrl gene in mice leads to functional cones at the expense of rods. We show that a 2.5-kb Nrl promoter segment directs the expression of enhanced GFP specifically to rod photoreceptors and the pineal gland of transgenic mice. GFP is detected shortly after terminal cell division, corresponding to the timing of rod genesis revealed by birthdating studies. In Nrl-/- retinas, the GFP+ photoreceptors express S-opsin, consistent with the transformation of rod precursors into cones. We report the gene profiles of freshly isolated flow-sorted GFP+ photoreceptors from wild-type and Nrl-/- retinas at five distinct developmental stages. Our results provide a framework for establishing gene regulatory networks that lead to mature functional photoreceptors from postmitotic precursors. Differentially expressed rod and cone genes are excellent candidates for retinopathies.\n",
    "\n",
    "### Overall Design\n",
    "> Postmitotic rod precursors and mature rod photoreceptors are tagged by GFP under the control of an Nrl promoter in the wild type background (Wt-Gfp mice). When cross-bred into the Nrl-knockout background (Nrl-ko-Gfp mice), the transformed “S-cones” are tagged by GFP. GFP positive photoreceptors from the Wt-Gfp or Nrl-ko-Gfp retina were enriched (purified) by FACS at five distinct developmental stages (E16, P2, P6, P10, and 4 weeks). Total RNA was extracted by Trizol regent and around 50 ng of total RNA was used for linear amplification and biotin labeling following Nugene kit protocol. Fragmented cDNA was hybridized on Affymetrix mouse genomic expression array 430 2.0 and then scanned with the standard protocol. Four replicates were performed for each time point."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "subslide"
    }
   },
   "source": [
    "# Loading the data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true,
    "slideshow": {
     "slide_type": "-"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "'data.frame':\t39 obs. of  4 variables:\n",
      " $ sidChar : chr  \"Sample_20\" \"Sample_21\" \"Sample_22\" \"Sample_23\" ...\n",
      " $ sidNum  : num  20 21 22 23 16 17 6 24 25 26 ...\n",
      " $ devStage: Factor w/ 5 levels \"E16\",\"P2\",\"P6\",..: 1 1 1 1 1 1 1 2 2 2 ...\n",
      " $ gType   : Factor w/ 2 levels \"wt\",\"NrlKO\": 1 1 1 1 2 2 2 1 1 1 ...\n",
      "'data.frame':\t29949 obs. of  39 variables:\n",
      " $ Sample_20: num  7.24 9.48 10.01 8.36 8.59 ...\n",
      " $ Sample_21: num  7.41 10.02 10.04 8.37 8.62 ...\n",
      " $ Sample_22: num  7.17 9.85 9.91 8.4 8.52 ...\n",
      " $ Sample_23: num  7.07 10.13 9.91 8.49 8.64 ...\n",
      " $ Sample_16: num  7.38 7.64 8.42 8.36 8.51 ...\n",
      " $ Sample_17: num  7.34 10.03 10.24 8.37 8.89 ...\n",
      " $ Sample_6 : num  7.24 9.71 10.17 8.84 8.54 ...\n",
      " $ Sample_24: num  7.11 9.75 9.39 8.37 8.36 ...\n",
      " $ Sample_25: num  7.19 9.16 10.11 8.2 8.5 ...\n",
      " $ Sample_26: num  7.18 9.49 9.41 8.73 8.39 ...\n",
      "  [list output truncated]\n"
     ]
    }
   ],
   "source": [
    "prDes <- readRDS(\"data/GSE4051_design.rds\")\n",
    "str(prDes)\n",
    "\n",
    "prDat<-read.table(\"data/GSE4051_data.tsv\",\n",
    "                  sep = \"\\t\", header = T, row.names = 1)\n",
    "str(prDat, list.len = 10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "subslide"
    }
   },
   "source": [
    "Here we examine the design of the study with a frequency table. We have four samples for each combination of development stage and genotype (except for \"E16 and NrlKO\", which has 3 samples)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "slideshow": {
     "slide_type": "-"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "         gType\n",
       "devStage  wt NrlKO\n",
       "  E16      4     3\n",
       "  P2       4     4\n",
       "  P6       4     4\n",
       "  P10      4     4\n",
       "  4_weeks  4     4"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "with(prDes, table(devStage, gType))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'WT samples = 20'"
      ],
      "text/latex": [
       "'WT samples = 20'"
      ],
      "text/markdown": [
       "'WT samples = 20'"
      ],
      "text/plain": [
       "[1] \"WT samples = 20\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "'NrlKO samples = 19'"
      ],
      "text/latex": [
       "'NrlKO samples = 19'"
      ],
      "text/markdown": [
       "'NrlKO samples = 19'"
      ],
      "text/plain": [
       "[1] \"NrlKO samples = 19\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Caculate total sample size of WT\n",
    "wtSample <- with(prDes, sum(gType == \"wt\"))\n",
    "paste(\"WT samples =\", wtSample)\n",
    "\n",
    "# Caculate total sample size of NrlKO\n",
    "nrlkoSample <- with(prDes, sum(gType == \"NrlKO\"))\n",
    "paste(\"NrlKO samples =\", nrlkoSample)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "subslide"
    }
   },
   "source": [
    "A cursory glance at the experession data reveals:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "slideshow": {
     "slide_type": "-"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Sample_20</th><th scope=col>Sample_21</th><th scope=col>Sample_22</th><th scope=col>Sample_23</th><th scope=col>Sample_16</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1415670_at</th><td> 7.236</td><td> 7.414</td><td>7.169</td><td> 7.070</td><td>7.383</td></tr>\n",
       "\t<tr><th scope=row>1415671_at</th><td> 9.478</td><td>10.020</td><td>9.854</td><td>10.130</td><td>7.637</td></tr>\n",
       "\t<tr><th scope=row>1415672_at</th><td>10.010</td><td>10.040</td><td>9.913</td><td> 9.907</td><td>8.423</td></tr>\n",
       "\t<tr><th scope=row>1415673_at</th><td> 8.362</td><td> 8.374</td><td>8.404</td><td> 8.487</td><td>8.363</td></tr>\n",
       "\t<tr><th scope=row>1415674_a_at</th><td> 8.585</td><td> 8.615</td><td>8.520</td><td> 8.641</td><td>8.509</td></tr>\n",
       "\t<tr><th scope=row>1415675_at</th><td> 9.591</td><td> 9.719</td><td>9.709</td><td> 9.700</td><td>9.656</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & Sample\\_20 & Sample\\_21 & Sample\\_22 & Sample\\_23 & Sample\\_16\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1415670\\_at &  7.236 &  7.414 & 7.169 &  7.070 & 7.383\\\\\n",
       "\t1415671\\_at &  9.478 & 10.020 & 9.854 & 10.130 & 7.637\\\\\n",
       "\t1415672\\_at & 10.010 & 10.040 & 9.913 &  9.907 & 8.423\\\\\n",
       "\t1415673\\_at &  8.362 &  8.374 & 8.404 &  8.487 & 8.363\\\\\n",
       "\t1415674\\_a\\_at &  8.585 &  8.615 & 8.520 &  8.641 & 8.509\\\\\n",
       "\t1415675\\_at &  9.591 &  9.719 & 9.709 &  9.700 & 9.656\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 5\n",
       "\n",
       "| <!--/--> | Sample_20 &lt;dbl&gt; | Sample_21 &lt;dbl&gt; | Sample_22 &lt;dbl&gt; | Sample_23 &lt;dbl&gt; | Sample_16 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| 1415670_at |  7.236 |  7.414 | 7.169 |  7.070 | 7.383 |\n",
       "| 1415671_at |  9.478 | 10.020 | 9.854 | 10.130 | 7.637 |\n",
       "| 1415672_at | 10.010 | 10.040 | 9.913 |  9.907 | 8.423 |\n",
       "| 1415673_at |  8.362 |  8.374 | 8.404 |  8.487 | 8.363 |\n",
       "| 1415674_a_at |  8.585 |  8.615 | 8.520 |  8.641 | 8.509 |\n",
       "| 1415675_at |  9.591 |  9.719 | 9.709 |  9.700 | 9.656 |\n",
       "\n"
      ],
      "text/plain": [
       "             Sample_20 Sample_21 Sample_22 Sample_23 Sample_16\n",
       "1415670_at    7.236     7.414    7.169      7.070    7.383    \n",
       "1415671_at    9.478    10.020    9.854     10.130    7.637    \n",
       "1415672_at   10.010    10.040    9.913      9.907    8.423    \n",
       "1415673_at    8.362     8.374    8.404      8.487    8.363    \n",
       "1415674_a_at  8.585     8.615    8.520      8.641    8.509    \n",
       "1415675_at    9.591     9.719    9.709      9.700    9.656    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(subset(prDat, select = 1:5))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "subslide"
    }
   },
   "source": [
    "# Subsetting the data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "subslide"
    }
   },
   "source": [
    "We extract the rows for each of these genes, transpose and then vectorize the data. We then merge this data together with the design metadata. Ensure that the samples match between the data and design before merging."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in data.frame(prDes, miniDat):\n",
      "\"row names were found from a short variable and have been discarded\""
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "'data.frame':\t78 obs. of  6 variables:\n",
      " $ sidChar : chr  \"Sample_20\" \"Sample_21\" \"Sample_22\" \"Sample_23\" ...\n",
      " $ sidNum  : num  20 21 22 23 16 17 6 24 25 26 ...\n",
      " $ devStage: Factor w/ 5 levels \"E16\",\"P2\",\"P6\",..: 1 1 1 1 1 1 1 2 2 2 ...\n",
      " $ gType   : Factor w/ 2 levels \"wt\",\"NrlKO\": 1 1 1 1 2 2 2 1 1 1 ...\n",
      " $ gene    : Factor w/ 2 levels \"Irs4\",\"Nrl\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      " $ gExp    : num  7.87 7.78 7.58 7.4 8.08 ...\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>sidChar</th><th scope=col>sidNum</th><th scope=col>devStage</th><th scope=col>gType</th><th scope=col>gene</th><th scope=col>gExp</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Sample_20</td><td>20</td><td>E16</td><td>wt   </td><td>Irs4</td><td>7.867</td></tr>\n",
       "\t<tr><td>Sample_21</td><td>21</td><td>E16</td><td>wt   </td><td>Irs4</td><td>7.783</td></tr>\n",
       "\t<tr><td>Sample_22</td><td>22</td><td>E16</td><td>wt   </td><td>Irs4</td><td>7.585</td></tr>\n",
       "\t<tr><td>Sample_23</td><td>23</td><td>E16</td><td>wt   </td><td>Irs4</td><td>7.400</td></tr>\n",
       "\t<tr><td>Sample_16</td><td>16</td><td>E16</td><td>NrlKO</td><td>Irs4</td><td>8.076</td></tr>\n",
       "\t<tr><td>Sample_17</td><td>17</td><td>E16</td><td>NrlKO</td><td>Irs4</td><td>7.515</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       " sidChar & sidNum & devStage & gType & gene & gExp\\\\\n",
       " <chr> & <dbl> & <fct> & <fct> & <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t Sample\\_20 & 20 & E16 & wt    & Irs4 & 7.867\\\\\n",
       "\t Sample\\_21 & 21 & E16 & wt    & Irs4 & 7.783\\\\\n",
       "\t Sample\\_22 & 22 & E16 & wt    & Irs4 & 7.585\\\\\n",
       "\t Sample\\_23 & 23 & E16 & wt    & Irs4 & 7.400\\\\\n",
       "\t Sample\\_16 & 16 & E16 & NrlKO & Irs4 & 8.076\\\\\n",
       "\t Sample\\_17 & 17 & E16 & NrlKO & Irs4 & 7.515\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| sidChar &lt;chr&gt; | sidNum &lt;dbl&gt; | devStage &lt;fct&gt; | gType &lt;fct&gt; | gene &lt;fct&gt; | gExp &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| Sample_20 | 20 | E16 | wt    | Irs4 | 7.867 |\n",
       "| Sample_21 | 21 | E16 | wt    | Irs4 | 7.783 |\n",
       "| Sample_22 | 22 | E16 | wt    | Irs4 | 7.585 |\n",
       "| Sample_23 | 23 | E16 | wt    | Irs4 | 7.400 |\n",
       "| Sample_16 | 16 | E16 | NrlKO | Irs4 | 8.076 |\n",
       "| Sample_17 | 17 | E16 | NrlKO | Irs4 | 7.515 |\n",
       "\n"
      ],
      "text/plain": [
       "  sidChar   sidNum devStage gType gene gExp \n",
       "1 Sample_20 20     E16      wt    Irs4 7.867\n",
       "2 Sample_21 21     E16      wt    Irs4 7.783\n",
       "3 Sample_22 22     E16      wt    Irs4 7.585\n",
       "4 Sample_23 23     E16      wt    Irs4 7.400\n",
       "5 Sample_16 16     E16      NrlKO Irs4 8.076\n",
       "6 Sample_17 17     E16      NrlKO Irs4 7.515"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "miniDat <- as.vector(t(prDat[c(\"1422248_at\", \"1450946_at\"), ]))\n",
    "miniDat <- data.frame(gene = rep(c(\"Irs4\", \"Nrl\"), each = nrow(prDes)),\n",
    "                      gExp = miniDat)\n",
    "miniDat <- data.frame(prDes, miniDat) # ignore the warning about row names\n",
    "str(miniDat)\n",
    "head(miniDat)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Code from Lecture 6"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Slide 5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "\tTwo Sample t-test\n",
       "\n",
       "data:  gExp by gType\n",
       "t = 0.52865, df = 37, p-value = 0.6002\n",
       "alternative hypothesis: true difference in means is not equal to 0\n",
       "95 percent confidence interval:\n",
       " -0.07383844  0.12597002\n",
       "sample estimates:\n",
       "   mean in group wt mean in group NrlKO \n",
       "           7.765750            7.739684 \n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "t.test(gExp ~ gType, miniDat, subset = gene == \"Irs4\", var.equal = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "            Df Sum Sq Mean Sq F value Pr(>F)\n",
       "gType        1 0.0066 0.00662   0.279    0.6\n",
       "Residuals   37 0.8764 0.02369               "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "summary(aov(gExp ~ gType, miniDat, subset = gene == \"Irs4\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "lm(formula = gExp ~ gType, data = miniDat, subset = gene == \"Irs4\")\n",
       "\n",
       "Residuals:\n",
       "     Min       1Q   Median       3Q      Max \n",
       "-0.36575 -0.09072  0.01725  0.09428  0.36525 \n",
       "\n",
       "Coefficients:\n",
       "            Estimate Std. Error t value Pr(>|t|)    \n",
       "(Intercept)  7.76575    0.03441 225.650   <2e-16 ***\n",
       "gTypeNrlKO  -0.02607    0.04931  -0.529      0.6    \n",
       "---\n",
       "Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1\n",
       "\n",
       "Residual standard error: 0.1539 on 37 degrees of freedom\n",
       "Multiple R-squared:  0.007497,\tAdjusted R-squared:  -0.01933 \n",
       "F-statistic: 0.2795 on 1 and 37 DF,  p-value: 0.6002\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "summary(lm(gExp ~ gType, miniDat, subset = gene == \"Irs4\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Slide 23"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "theHit <- which(rownames(prDat) == \"1440645_at\") # 17843\n",
    "## and this as our boring gene\n",
    "theBore <- which(rownames(prDat) == \"1443184_at\") # 18898\n",
    "\n",
    "keepers <- data.frame(row = c(theBore, theHit),\n",
    "                       probesetID = I(rownames(prDat)[c(theBore, theHit)]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in data.frame(prDes, miniDat):\n",
      "\"row names were found from a short variable and have been discarded\""
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "'data.frame':\t78 obs. of  6 variables:\n",
      " $ sidChar : chr  \"Sample_20\" \"Sample_21\" \"Sample_22\" \"Sample_23\" ...\n",
      " $ sidNum  : num  20 21 22 23 16 17 6 24 25 26 ...\n",
      " $ devStage: Factor w/ 5 levels \"E16\",\"P2\",\"P6\",..: 1 1 1 1 1 1 1 2 2 2 ...\n",
      " $ gType   : Factor w/ 2 levels \"wt\",\"NrlKO\": 1 1 1 1 2 2 2 1 1 1 ...\n",
      " $ gene    : Factor w/ 2 levels \"theBore\",\"theHit\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      " $ gExp    : num  7.59 7.96 7.43 7.58 7.98 ...\n"
     ]
    }
   ],
   "source": [
    "miniDat <- as.vector(t(prDat[keepers$probesetID, ]))\n",
    "miniDat <- data.frame(gene = rep(c(\"theBore\", \"theHit\"), each = nrow(prDes)),\n",
    "                      gExp = miniDat)\n",
    "\n",
    "miniDat <- data.frame(prDes, miniDat)\n",
    "str(miniDat)\n",
    "\n",
    "boreDat <- filter(miniDat, gene == \"theBore\")\n",
    "hitDat <- filter(miniDat, gene == \"theHit\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAMAAADKOT/pAAAAV1BMVEUAAAADAwMHBwcODg4P\nDw8dHR0fHx8zMzM6Ojo/Pz9NTU1oaGh1dXV8fHx/f3+MjIyampqnp6eysrK9vb3Hx8fQ0NDZ\n2dnh4eHp6enr6+vw8PD/AAD////hsE7hAAAACXBIWXMAABJ0AAASdAHeZh94AAAgAElEQVR4\nnO2dbUPa2roAOff0WMHal713t63l///OSwLyEgLmSRZZg8/MB6WIcZpkXMkiwmItIpNZ1BYQ\n+QgYkkgBDEmkAIYkUgBDEimAIYkUwJBECmBIIgUwJJECGJJIAUaH9Pq9/fYL37/Y8vTX2MWL\n3BWjQ9om9E5Ii8Xz2OWL3BM3C6n99PK8+DH2B4jcEbcNaf1rsRr7A0TuiLEhtcdtbTB/rxbP\n/7b3/fu8OZT7ufvy/nHtp5fvq8Xq+8v2ntenxbfTh4vcOZND+t7eakr6Z3tW9GP75e3DXhdP\nzaefuzOmn+2XvrUPOn64yJ0z+dBu9XP9+q2ZU3hpB6bNWdHP9fE5UvOvl8Xir9f164/F4lfz\npefXdefhInfO5JCaDl6bf/xYNH1sbn9bH83atfPfP3bjzvfm83b4On24yJ1TZrKh+fS0j2d9\nFNK35rzoqRmJ1s3Uw9P+W04eLnLnlAtpcRpSe//r36tm/DmZeth/iyHJx6FkSGdfXbenQpdC\nGvuTRXiUC2m1eOl+9e1W36HdycNF7pxyIX1ftFfftUPQ8Yi0aiYW2i+9TTbsbh89XOTOmRDS\nr/VJSC/tc0Ivq+Pp7/W/q+bOX5svbae/Xw4z48cPF7lzRof0tOie8Pw8eUL25KrV0ydktws4\nfrjInTM6pJen5qDtZObg14/V4unv3R272e9/do8+vkRot4Sjh4vcOc6diRTAkEQKYEgiBTAk\nkQIYkkgBDEmkAIYkUgBDEimAIYkUwJBECmBIIgUYG9JnEfk8PaSR3/cuv2+14CkgpZhWuaQM\nKQZSimmVS8qQYiClmFa5pAwpBlKKaZVLypBiIKWYVrmkDCkGUopplUvKkGIgpZhWuaQMKQZS\nimmVS8qQYiClmFa5pAwpBlKKaZVLKhbS9t0rVxv6llCYXBtiEkirXFKhkLb9HD50llCYXBti\nEkirXFKRkFZrQ0JKMa1ySY04tDMkHEirXFKTQtpeP/5bJD2OSDGQUkyrXFKGFAMpxbTKJWVI\nMZBSTKtcUoYUAynFtMolZUgxkFJMq1xSXtkQAynFtMol5bV2MZBSTKtcUoYUAynFtMolZUgx\nkFJMq1xShhQDKcW0yiVlSDGQUkyrXFKGFAMpxbTKJWVIMZBSTKtcUoYUAynFtMolZUgxkFJM\nq1xShhQDKcW0yiVlSDGQUkyrXFKGFAMpxbTKJWVIMZBSTKtcUoYUAynFtMolZUgxkFJMq1xS\nhhQDKcW0yiVlSDGQUkyrXFKGFAMpxbTKJWVIMZBSTKtcUoYUAynFtMolZUgxkFJMq1xShhQD\nKcW0yiVlSDGQUkyrXFIFQqr9PgAi9XFEioGUYlrlkjKkGEgpplUuKUOKgZRiWuWSMqQYSCmm\nVS4pQ4qBlGJa5ZIypBhIKaZVLilDioGUYlrlkjKkGEgpplUuKUOKgZRiWuWSMqQYSCmmVS4p\nQ4qBlGJa5ZIypBhIKaZVLilDioGUYlrlkjKkGEgpplUuKUOKgZRiWuWSMqQYSCmmVS4pQ4qB\nlGJa5ZIypBhIKaZVLilDioGUYlrlkjKkGEgpplUuKUOKgZRiWuWSMqQYSCmmVS4pQ4qBlGJa\n5ZIypBhIKaZVLqkxIa029C2hMLk2xCSQVrmkRoS02n/oLKEwuTbEJJBWuaQMKQZSimmVS8qQ\nYiClmFa5pCaF9Lml9suXi9RnZEgrRyQSSKtcUiNn7Ty0Q4G0yiU19nkkQyKBtMol5WRDDKQU\n0yqXlCHFQEoxrXJJeWVDDKQU0yqXlNfaxUBKMa1ySRlSDKQU0yqXlCHFQEoxrXJJGVIMpBTT\nKpeUIcVASjGtckkZUgykFNMql5QhxUBKMa1ySRlSDKQU0yqXlCHFQEoxrXJJGVIMpBTTKpeU\nIcVASjGtckkZUgykFNMql5QhxUBKMa1ySRlSDKQU0yqXlCHFQEoxrXJJGVIMpBTTKpeUIcVA\nSjGtckkZUgykFNMql5QhxUBKMa1ySRlSDKQU0yqXlCHFQEoxrXJJGVIMpBTTKpeUIcVASjGt\nckkVCKn2+wCI1McRKQZSimmVS8qQYiClmFa5pAwpBlKKaZVLypBiIKWYVrmkDCkGUopplUvK\nkGIgpZhWuaQMKQZSimmVS8qQYiClmFa5pAwpBlKKaZVLypBiIKWYVrmkDCkGUopplUvKkGIg\npZhWuaQMKQZSimmVS8qQYiClmFa5pAwpBlKKaZVLypBiIKWYVrmkDCkGUopplUvKkGIgpZhW\nuaQMKQZSimmVS8qQYiClmFa5pAwpBlKKaZVLypBiIKWYVrmkDCkGUopplUvKkGIgpZhWuaQM\nKQZSimmVS2pMSKsNfUsoTK4NMQmkVS6pESGt9h86SyhMrg0xiYpWy+XywleQqypjSI8XN1FF\nkHtHRavl169fL2wm5KpKGNLyy9kmuvzbbzaQe0c9q6ajSyUhVxUzpM8tt3l1/8cvDY/dux4v\nPV5q0LOVkoKdbFh+/dL5XXftt99sIH/NOiINhjkidZZQFEMK4DnSUPKFdH6OZEgXcdZuKAlD\nOp+1u/LbbzaQewfTKpcUOKTz/7OzdhdwVQ2FFNIHv7Lh+i6I3DvOrXIP3lc24c2enPRauw7v\n7IL3EVLu08krm7DnyclCGNIp7+2ChjSYWqvqyn/+fCq4GIZ0iiEVw5DGLqEwhjSI5oTAc6Qj\naCG9/nhaLJ7/CiyhMJ4jDaH1ddbuGNY50q/VomX1a/ASCuOs3QC2v34fa2v04axdy/PieZPQ\nr+fF98FLKAxtn22hSRlSjPmfR1ps//26eO/cyZBqYkgx5g/p2+J1e/fz4CUUJteGGMuFc6Ta\nbI6ueFLrKlc2fH9+aQ7tnrOdI12HJ3U0aweYYdixTDdMXjm0O2bQEgrD22fXUKmdFWHOe0t7\nwPmF4XKKIUFASm2tEM/CbjGkSUsoDGefPTpk4kgdY0hDMaSKHB8yYaROoIVEPUdaLv87//NI\nL9vPr98GL6EwvyEnzyc7KDgk0DkSdNZu+fXh8WHuKxsWix/Np79qPY+0XD5W2zH2Abc37iYk\nyC+elmWz/TA2WzYdbaRuVNLFkF6eF6t//lktnl4GL6Ekm733fw+VDlX2AW9v3E9IHDbr7OF/\nnPFxS52Q1utNRYunn4ElFKTZef+3fKgS0j6ctxudcyTQ7/036l0f2r8umo6Wm6MoVkm1QvqZ\nLKQ/MxM3vJhwwZDmXguDKPR/q3KO9PJU89CuCenLw5BfaeW31tmIdMzvzcZ4KLMxypmXJPI/\nuDhPWHpEKvffO5+1K3SAcWWyof1TpFqTDZtNUX7DD//Zx+dI2/va1d2cQT9sbj7UOGS5Mr1d\n6y9OLhohz5EabvYnkJcvWt1es1pt+nuzz95mwYN+9rJ74+1k6ct/lg2faljtR8qz36G4kJCz\ndg03+6N8n5AdwnZ1f2qPN/+v3oi0Pajs+R1a7a+6N0YXVwVo+x2YOaSj47n3D+0mvIT/4+N9\nvJHB9k0X/tt8+F9Dnfdf+O9mdf0X9QYQj5t1wTAZS7G1eSWkbUK3PEe6fnwK+o12NCK1H+s8\nTbwdkXqnQGr4vPfrHLT9DrxJdQ7ap1MzpHeGVdKGOJwjLas9j7RbXYY0hbdrQLrTSJMxpGHs\nZ+0qKvQ8P7yDGRLwmeujqxILH1cYUoyaUm8FYWbtrh8XPYKuod3zIUO6n3OkA1Wl5riyIcaV\nQad5CTleSR8zpOvHp4Y0mH6ruodWqJD2q+L8HKkQdUO6QtXTkcsgX/Ww32r2v0I5NSSFdFgV\nZ7N2pX7fXApp2Os1nC6hJMtsf6r8Hlez6LOa/Q9mu4acc6SjVdHz4s63fUK2ckjtRauQzXAC\n8JXh14yQzn4eZ9buWkilVhP0EiFD6nCPIc32o9/DkG6x6GncT0hznyOBQ+o5Rzr50k1DOhza\nrb5ffbFVz5FmIXyONPusXdcQFNLZrN3xl2589ffJadK1kpy1m4fwrN3sdAwZUh16pG45a7fh\n59vbuvyz/nH1rV0y/BnFAaQU0yqX1MWQnt7ejeLpneeSDKk+SKtcUu+9P1L72ZD2IKWYVrmk\n3n/Hvuf1v82oNGAJhcm1ISaBtMolNeQ9ZBeLvwctoTC5NsQkkFa5pK68q/lfT4vF04/NmdL2\nxYvfX0Jhcm2ISSCtcklBn5BtyLUhJoG0yiVlSDGQUkyrXFKGFAMpxbTKJWVIMZBSTKtcUoYU\nAynFtMolNSKkVUvPEgqTa0NMAmmVS2rsiGRIJJBWuaRGhnToyJAAIK1ySRlSDKQU0yqX1LiQ\ndh19bpn+AuQi986kkDopTqHnr6ty/UabBNIqlxQkpL6/9821ISaBtMolNSqk446KhNT7ChS5\nNsQkkFa5pAwpBlKKaZVLypBiIKWYVrmkGCF5jjQNpFUuKcq1ds7aTQFplUuKElIPuTbEJJBW\nuaQMKQZSimmVS8qQYiClmFa5pAwpBlKKaZVLypBiIKWYVrmkDCkGUopplUvKkGIgpZhWuaQM\nKQZSimmVS8qQYiClmFa5pAwpBlKKaZVLypBiIKWYVrmkDCkGUopplUvKkGIgpZhWuaQMKQZS\nimmVS8qQYiClmFa5pAwpBlKKaZVLypBiIKWYVrmkDCkGUopplUvKkGIgpZhWuaQMKQZSimmV\nS8qQYiClmFa5pAwpBlKKaZVLqkBItd8HQKQ+jkgxkFJMq1xShhQDKcW0yiVlSDGQUkyrXFKG\nFAMpxbTKJWVIMZBSTKtcUoYUAynFtMolZUgxkFJMq1xShhQDKcW0yiVlSDGQUkyrXFKGFAMp\nxbTKJWVIMZBSTKtcUoYUAynFtMolZUgxkFJMq1xShhQDKcW0yiVlSDGQUkyrXFKGFAMpxbTK\nJWVIMZBSTKtcUoYUAynFtMolZUgxkFJMq1xShhQDKcW0yiVlSDGQUkyrXFKGFAMpxbTKJWVI\nMZBSTKtcUoYUAynFtMolZUgxkFJMq1xSY0JabehbQmFybYhJIK1ySY0IabX/0FlCYXJtiEkg\nrXJJGVIMpBTTKpfU2JD6llCYXBtiEkirXFKjQno7R/rcUvt9AETqMyak1dpDOxZIq1xSniPF\nQEoxrXJJGVIMpBTTKpeUIcVASjGtckkZUgykFNMql5RXNsRASjGtckl5rV0MpBTTKpeUIcVA\nSjGtckkZUgykFNMql5QhxUBKMa1ySRlSDKQU0yqXlCHFQEoxrXJJGVIMpBTTKpeUIcVASjGt\nckkZUgykFNMql5QhxUBKMa1ySRlSDKQU0yqXlCHFQEoxrXJJGVIMpBTTKpeUIcVASjGtckkZ\nUgykFNMql5QhxUBKMa1ySRlSDKQU0yqXlCHFQEoxrXJJGVIMpBTTKpeUIcVASjGtckkZUgyk\nFNMql5QhxUBKMa1ySRUIqfbLl4vUxxEpBlKKaZVLypBiIKWYVrmkDCkGUopplUvKkGIgpZhW\nuaQMKQZSimmVS8qQYiClmFa5pAwpBlKKaZVLypBiIKWYVrmkDCkGUopplUvKkGIgpZhWuaQM\nKQZSimmVS8qQYiClmFa5pAwpBlKKaZVLypBiIKWYVrmkDCkGUopplUvKkGIgpZhWuaQMKQZS\nimmVS8qQYiClmFa5pAwpBlKKaZVLypBiIKWYVrmkDCkGUopplUvKkGIgpZhWuaQMKQZSimmV\nS8qQYiClmFa5pAwpBlKKaZVLakxIq4aeJRQm14aYBNIql9SokC4soTC5NsQkkFa5pAwpBlKK\naZVLakRIpx0ZEgCkVS6pMSHtT5E+t9R+HwCR+owdkZxsIIG0yiU1dvrbkEggrXJJGVIMpBTT\nKpeUh3YxkFJMq1xSI0M6mrkzpPogrXJJjb2yoW8Jhcm1ISaBtMol5bV2MZBSTKtcUoYUAynF\ntMolZUgxkFJMq1xShhQDKcW0yiVlSDGQUkyrXFKGFAMpxbTKJWVIMZBSTKtcUoYUAynFtMol\nVSAkEZke0s242VA3BaQU0yqrlCENASnFtMoqZUhDQEoxrbJK4UISuUcMSaQAhiRSAEMSKYAh\niRTAkEQKgAhptWP7j/1dNZXWe4udCEOoBWd1vPmqbr+BP/IWZoyQjm8fXq6o/i5y2DcgQi04\nq+OEam6/oe1mCGm1NqT3wFkdhVRz+60ckc7+UX/naDk5WgFa4YTW63ohna6PE43OkXD37hJg\nQ2Ic/J984u23xNUECmnVGSSPP5cezxkhnU02HH5xVOXkLBrgs+VgRVxNtULqrorVLqFOQEd3\nFfVjhHT2D8jx3erKv+qxOr1VX4uw/c5+4nFIu873n98Gz4KGhnSNczEChtTncHLwsD45ruse\nUqz20xIJzpG6d1dhdeF2XQxpiMi1kNbn903FkK7RnQViYEjDRQ7R5JxsoDxlf7i1Yhg1HOeN\nkOrbf+tf2XAyd5Vj+lvk3jEkkQIYknwoVmfTd/NgSCIFMCSRAhiSSAEMSaQAhiRSAEMSKYAh\niRTAkEQKYEgiBTAkkQIYkkgBDEmkAIYkUgBDEimAIYkUwJBECmBIIgUwJJECGJJIAQxJpACG\nJFIAQxIpgCGJFMCQRApgSCIFMCSRAhiSSAEMSaQAhiRSAEMSKYAhiRTAkEQKYEgiBTAkkQIY\nkkgBDEmkAIYkUgBDEimAIYkUwJBECmBIIgUwJJECjA7p9Xv77Re+/3B/e6v9sP0OkY/I6JC2\nqURCuvRYkftnnpDW1x4rcv8YkkgBxu7ci4Y2jr9Xi+d/2/v+fV4snn/uvnx44O7D7jtEPiKT\nQ/re3mpK+qe9tfix/fLhgWtDkg/P5EO71c/167fF83r90g5ML8+Ln+vekDy0kw/M5JCabF6b\nf/xYvK7b29/Wu+Fnx9qQ5MNTZrKh+fR0Wo4hSSbKhXRezuFxhiQfnJIhnX11f8uQ5INTLqTV\n4qX71f0tQ5IPTrmQvi/aa+lemhk8Q5JkTAjp1/okpJf2KaSX1ZXp718TZUWojA7paXF0YtR+\n+nn9Cdndd4h8REbv2i9Pi9VpSOtfP1aLp7+P7tjfaj9sv0PkI+IYIVIAQxIpgCGJFMCQRApg\nSCIFMCSRAhiSSAEMSaQAhiRSAEMSKYAhiRRgbEifReTz9JBGft+7/L7VgqeAlGJa5ZIypBhI\nKaZVLilDioGUYlrlkjKkGEgpplUuKUOKgZRiWuWSMqQYSCmmVS4pQ4qBlGJa5ZIypBhIKaZV\nLilDioGUYlrlkjKkGEgpplUuqVhI2xfRWm3oW0Jhcm2ISSCtckmFQtr2c/jQWUJhcm2ISSCt\ncklFQlqtDQkpxbTKJTXi0M6QcCCtcklNCml7/fhvkfQ4IsVASjGtckkZUgykFNMql5QhxUBK\nMa1ySRlSDKQU0yqXlCHFQEoxrXJJeWVDDKQU0yqXlNfaxUBKMa1ySRlSDKQU0yqXlCHFQEox\nrXJJGVIMpBTTKpeUIcVASjGtckkZUgykFNMql5QhxUBKMa1ySRlSDKQU0yqXlCHFQEoxrXJJ\nGVIMpBTTKpeUIcVASjGtckkZUgykFNMql5QhxUBKMa1ySRlSDKQU0yqXlCHFQEoxrXJJGVIM\npBTTKpeUIcVASjGtckkZUgykFNMql5QhxUBKMa1ySRlSDKQU0yqXlCHFQEoxrSpJLZfLK181\nJAhIKaZVHanl169fr5T0+53QRmNIMZBSTKsqUk1H10r6/U5ooykQUu33ARDZ8/il4XH010fj\niBQDKcW0Qo5Ij+98fTSGFAMpxbRCniMZEgSkFNOKOWt3ElrBiQdDioGUYlpBpY7iKTnxYEgx\nkFJMK6rUvqT3zqdCGFIMpBTTCiq1yefhU1vPIaQCh3iGFAMpxbRiSjUdLZcPbT5vIZU4xDOk\nGEgpphVTatl0tHz4+lbSstAhniHFQEoxrZhSxyHtDukMqQJIKaYVVGrZHNmddGNIFUBKMa2o\nUstPD51sPEeaH6QU04ordTZJ56zd7CClmFa5pAwpBlKKaZVLypBiIKWYVrmkDCkGUopplUvK\nkGIgpZhWuaQMKQZSimmVS8qQYiClmFa5pAwpBlKKaUWU+mNIEJBSTCug1B9HJApIKaYVT+qP\nh3YYkFJMK5zUpiNDooCUYlrlkhoT0mpD3xIKk2tDTAJpRZNqBiRUSKv9h84SCkPbEC1IKaYV\nTKrtyJAoIKWYViypbUeGRAEpxbRCSe06gob0uaX8C5KLlOXPzX/CyJBWjkgkkFYgqT/7W6QR\nqZm189AOBdKKI3XoiBVSgyGRQFphpI46QoXkZAMOpBVF6rgjQ6KAlGJa5ZLyyoYYSCmmFUTq\nZEBihXRpCYWBbIhTkFJMK4bUaUeGRAEpxbRCSHU6MiQKSCmmFUGq25EhUUBKMa0AUmcdGRIF\npBTTqr7UeUeGRAEpxbSqLtXTkSFRQEoxrWpL9XVkSBSQUkyrXFKGFAMpxbSqLNU7IBkSBaQU\n06quVH9HhkQBKcW0qip1oSNDooCUYlrVlLrUkSFRQEoxrSpKXezIkCggpZhW9aQud2RIFJBS\nTKtqUlc6MiQKSCmmVS2pax2dSS2XyzI/1ZBiIKWYVpWkrnbUlVp+/fq1TEmGFAMpxbS6A6mm\no0IlGVIMpBTTqo7U9QHJkCggpZhWVaTe6ciQKCClmFY1pN7ryHMkCkgpplUFqXc7ctaOAlKK\naTW/1PsdoZ9HuvkL/YsM4fZvOXEFR6QYSCmm1dxSA8Yj9ohURuQc947BIK1mlhrUkSFRQEox\nreaVGtaRIVFASjGtAFLnc3KGBAEpxbSaVap3QOp5lsiQICClmFZzSl3sqFuSIUFASjGtZpTq\nP0EypC3Z944ASKv5pC5MNBjSluR7RwSk1WxSFyfsPEdqyb13hEBazSV1ZeLbWbuG1HtHDKTV\nTFIDn0DaYUgQkFJMq3mkYh0ZEgWkFNNqFqlgR4ZEASnFtMolZUgxkFJMqzmkogOSIVFASjGt\nZpAKd2RIFJBSTKvbS8U7MiQKSCmm1c2lRnRkSBSQUkyrW0uN6ciQKCClmFY3lhrVkSFRQEox\nrW4rNa4jQ6KAlGJa3VRqZEeGRAEpxbTKJWVIMZBSTKtbSo0dkAyJAlKKaXVDqdEdsUJabehb\nQmGy7R0TQFrdTmp8R6iQVvsPnSUUJtneMQWk1c2kJnRkSBSQUkyrW0lN6ciQKCClmFY3kprU\nETSkzy013wNAslH1LSeu4GRDDKQU0+omUtPGI+qI1FlCYfLsHZNBWt1CampHhkQBKcW0yiVl\nSDGQUkyrG0hNHpAMiQJSimlVXmp6R6iQnGzAgbQqLlWgI1ZIl5ZQmBx7RxGQVqWlSnRkSBSQ\nUkyrwlJFOjIkCkgpplVZqTIdGRIFpBTTqqhUoY4MiQJSimlVUqpUR4ZEASnFtCooVawjQ6KA\nlGJa5ZIypBhIKaZVOalyA5IhUUBKMa2KSRXsyJAoIKWYVqWkSnZkSBSQUkyrQlJFOzIkCkgp\nplUZqbIdGRIFpBTTqohU4Y4MiQJSimlVQqp0R4ZEASnFtCogVbwjQ6KAlGJaTZcq35EhUUBK\nMa1ySRlSDKQU02qy1A0GJEOigJRiWk2VukVHhkQBKcW0mih1k44MiQJSimk1Teo2HRkSBaQU\n02qS1I06MiQKSCmm1RSpW3WEDqn2+wDIh4P6lhNXcESKgZRiWo2Xutl4xB6Ryoic88H2jluC\ntBotdcOODIkCUopplUvKkGIgpZhWY6VuOSAZEgWkFNNqpNRNOzIkCkgpptU4qdt2ZEgUkFJM\nq1FSN+7IkCggpZhWY6Ru3ZEhUUBKMa1GSN28I0OigJRiWsWlbt+RIVFASjGtwlIzdGRIFJBS\nTKtcUoYUAynFtIpKzTEgGRIFpBTTKig1S0eGRAEpxbSKSc3TkSFRQEoxrUJSM3VkSBSQUkyr\niNRcHRkSBaQU0yogNVtHhkQBKcW0Gi41X0eGRAEpxbQaLDVjR4ZEASnFtBoqNWdHhkQBKcW0\nyiVlSDGQUkyrgVKzDkiGRAEpxbQaJjVvR6iQVi09SyjMHe8dc4O0GiQ1c0eokFoMiQTSaojU\n3B3hQjp0ZEgAkFYDpGbvyJAoIKWYVu9Lzd8RLaRdR59bar98udwpd/hS+ZeZFFInxcLc6a/Z\nGiCt3pOqMB5RR6TOEgpzl3tHHZBW70hV6QgW0nFHhgQAaZVLypBiIKWYVtel6gxIhkQBKcW0\nuipVqSNDooCUYlpdk6rVESykC0sozN3tHfVAWl2RqtaRIVFASjGtLkvV68iQKCClmFYXpSp2\nZEgUkFJMq0tSNTsyJApIKabVBamqHRkSBaQU06pfqm5HhkQBKcW0yiVlSDGQUkyrXqnKA5Ih\nUUBKMa36pGp3ZEgUkFJMqx6pTUfL5XJ+lQOGBAEpxbQ6l2o6+vr1a82SDAkCUoppdSa166hq\nSYYEASnFtOpKNedHhjRkCYW5i72DAdKqI9XOMxjSkCUU5h72Dggoq7fphFOp3Xyd50gDllAY\n1N7xBlIKZbVPpV/KWbv3l1AY0t6xBylFsjocvJ1IVX8CaYchQUBKkaz6Q6J0ZEgUkFIkq96Q\nMB0ZEgWkFMqq5xyJ05EhUUBKsazOZu1AHRkSBaQU0+pNitSRIVFASjGtdlLnHdWcASeHVPt9\nAITM+VtOPH758uWxgsltcUSKgZRiWrVSPeNR1auEyCNSGZFzuHsHDqTVBSlDencJhbmnvaMy\nSKtGqmeiwZDeXUJhsHsHD6TV7wsTdlWvWzUkCEgpptXvSxPfztq9s4TCQPcOIkir36gnkHYY\nEgSkFNHqzx9iR4ZEASlFs/rTVgST2mJIEJBSJKs/+6FoJ1X5L/k6GBIEpBTCahPMn5Pjua1U\n7b8t72BIEJBSAKumoT+nwbRS1V/tpIMhQUBKVbZqGzoPxpDGLqEwyH0WKVXR6m0guhbSw8On\n/Z21z5gMCcKZVO09o6XOqmpPiXbnQJdC2nzhYfmw/0L1MyZDgtCVqrlnHBqefVX9eZtX2Ad0\nviJ2Up8eHvZfqn+gZ0gQOlI194yjXXfWVXUyN3dYAWdD8+/uI6KwgzAAAAdmSURBVAxp4BIK\nY0hXOf7Rs62qP90LFq6sAEMau4TCGNJVZg/pLKK9Rv///3fPIzxHGrKEwtxDSBX3jDlD+tMf\n0dbj0mzL775H1J6bMSQIoFm7mc6RrjR0nfvYfqUwpBgkqdvP2o2OqIG0qvYYEgSk1E2sJkXU\ngFxVhnSg6mE2cu8obXXtlGg4yFVlSHvqTvwg946SVkUaakGuKkN6o/JTEci9o5RVuYgakKsK\nFdJqQ98SCmNIg5luVeZo7gTkqiKFtNp/6CyhMIY0mGlW5RtqQa4qQ9rjOdI5461uFFEDclXh\nQupbQmGctRvMOKsbRtSAXFWskN7OkT631H4fgFvz+Pjx3juhPSWqLfGhGBPSal3z0G5uTg8l\nIVIdei9ru3TPjQeiPchVxRqR1olC6kxuMKS69F1o3XvPXBEdS6EwpFrcU0jnM5rH98wZ0ZEU\nC0OqxYcI6QbPEg2WYmFIs/N2cnFH50i9IdVo6FiKBSmkulc2zMWhn5PTdeTecfEcqYmo2rNu\nyFWFCunSEgpTdUNcuoACsHf0PI/WN2u3G4nqPesGWFXnGNLMcEPqu7KjY1XllOiM+quqB0Oa\nmZPXvlkepr0uUkNsz9GqQjTUYkhjl1AYzDnScSsXpa5WVjC9ayFxImowpLFLKEz1Wbu3Pf5h\nMyS9vXp14T+hi6d3KSRWRA2GNHYJham2IU6HjGXT0fZDI1Xzktny49stMaSxSyjM3Bviwj65\nC+nTdhx4rP0Kh9dn7UjkkjKk9Xu/0z81HT38X/Na8Mvll9qvudvLyaqq/RqMbxjS2CUUZpaX\nDx1yUNS8n8J/mpruIqTGFiFoSGOXUJgb/Z9HnFUsl5/eTpTwIX3aFl/PZY8hjV1CYW7xYm1j\nz8qbt8tqp+4A50g9HFbV29RIRZk3DGnsEgpT8DWmJk9rLZs3cGx2z6qzdhcxpKEYUpiy88Jv\nz88i947jkL4eP+tVFeSqyhBS93f9+Ff0uMUTKzs75N5xbLU8ee/jmiBXVYKQzi7GjP2fZ3pi\nErl3nFhhjj2Rq+rjh3R+6cuw//PMz+wj9w6mVS6pew2p1qUxyL2DaZVL6t5Cqn1pGXLvYFrl\nksKE9M45Uu2A3kDuHUyrXFKckHpn7ZaUgN5A7h1Mq1xSoJC6tAHRriNA7h3HzyN1fhtVnMND\nrqqMIf2u/RYufSD3jqMXP+l5XTtf/OQIckjTXnz82kvUP35p+HgvYX8zuivMFTgbtUekK78y\nHZEGc/y6dscrrOoKRK4q8og05cdf29K/L2fmi7WdYkhDSRrShWIqHvkj9w7PkQaTNaTTBx+9\nGnet/QO5dzhrN5iPGtL1c6QLDzWkLkirXFK1Q+r9lbl75avOnft8DKkL0iqXVPWQetgNPRdD\n8hypC9IqlxQwpLdiLofkrF0HpFUuqfsJqeoM1B7k3sG0yiV1RyEh/vYTuXcwrXJJAUO6cI7E\nACnFtMolRQypf9aOAVKKaZVLChnSllwbYhJIq1xShhQDKcW0yiVlSDGQUkyrXFKGFAMpxbTK\nJWVIMZBSTKtcUoYUAynFtMolZUgxkFJMq1xShhQDKcW0yiVlSDGQUkyrXFKGFAMpxbTKJWVI\nMZBSTKtcUoYUAynFtMolZUgxkFJMq1xSY0JaNfQsoTC5NsQkkFa5pEaFdGEJhcm1ISaBtMol\nZUgxkFJMq1xSI0I67ciQACCtckmNCWl/ivS5pfb7AIjUZ+yI5GQDCaRVLqmx09+GRAJplUvK\nkGIgpZhWuaQ8tIuBlGJa5ZIaGdLRzJ0h1QdplUtq7JUNfUsoTK4NMQmkVS4pr7WLgZRiWuWS\nMqQYSCmmVS4pQ4qBlGJa5ZIypBhIKaZVLilDioGUYlrlkjKkGEgpplUuKUOKgZRiWuWSMqQY\nSCmmVS6pAiGJyPSQbsbNhropIKWYVlmlDGkISCmmVVYpQxoCUopplVUKF5LIPWJIIgUwJJEC\nGJJIAQxJpACGJFIAREirHdt/7O+qqbTeW+xEGEItOKvjzVd1+w38kbcwY4R0fPvwckX1d5HD\nvgERasFZHSdUc/sNbTdDSKu1Ib0HzuoopJrbb+WIdPaP+jtHy8nRCtAKJ7Re1wvpdH2caHSO\nhLt3lwAbEuPg/+QTb78lriZQSKvOIHn8ufR4zgjpbLLh8IujKidn0QCfLQcr4mqqFVJ3Vax2\nCXUCOrqrqB8jpLN/QI7vVlf+VY/V6a36WoTtd/YTj0Padb7//DZ4FjQ0pGucixEwpD6Hk4OH\n9clxXfeQYrWflkhwjtS9uwqrC7frYkhDRK6FtD6/byqGdI3uLBADQxoucogm52QD5Sn7w60V\nw6jhOG+EVN/+W//KhpO5qxzT3yL3jiGJFMCQ5EOxOpu+mwdDEimAIYkUwJBECmBIIgUwJJEC\nGJJIAQxJpACGJFIAQxIpgCGJFMCQRApgSCIFMCSRAhiSSAEMSaQAhiRSAEMSKYAhiRTAkEQK\nYEgiBTAkkQIYkkgBDEmkAIYkUgBDEimAIYkUwJBECmBIIgUwJJECGJJIAQxJpAD/DyBuzbK9\nd8VTAAAAAElFTkSuQmCC",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "boreLim <- ggplot(boreDat, aes(x = devStage, y = gExp)) + \n",
    "             geom_jitter(width = 0.2, alpha = 0.5) +\n",
    "             labs(title = \"theBore\") +\n",
    "             theme_bw() +\n",
    "             theme(legend.position = \"none\") +\n",
    "             ylim(5, 10) +\n",
    "             xlab(\"\") +\n",
    "             stat_summary(aes(group=1), fun.y=mean, geom=\"line\", colour=\"red\")\n",
    "\n",
    "hitLim <- ggplot(hitDat, aes(x = devStage, y = gExp)) + \n",
    "             geom_jitter(width = 0.2, alpha = 0.5) +\n",
    "             labs(title = \"theHit\") +\n",
    "             theme_bw() +\n",
    "             theme(legend.position = \"none\") +\n",
    "             ylim(5, 10) +\n",
    "             ylab(\"\") +\n",
    "             xlab(\"\") +\n",
    "             stat_summary(aes(group=1), fun.y=mean, geom=\"line\", colour=\"red\")\n",
    "\n",
    "grid.arrange(boreLim, hitLim, nrow = 2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Slide 24"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A grouped_df: 4 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gType</th><th scope=col>gene</th><th scope=col>mean(gExp)</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>wt   </td><td>theBore</td><td>7.643900</td></tr>\n",
       "\t<tr><td>wt   </td><td>theHit </td><td>6.528350</td></tr>\n",
       "\t<tr><td>NrlKO</td><td>theBore</td><td>7.477421</td></tr>\n",
       "\t<tr><td>NrlKO</td><td>theHit </td><td>6.612632</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped_df: 4 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       " gType & gene & mean(gExp)\\\\\n",
       " <fct> & <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t wt    & theBore & 7.643900\\\\\n",
       "\t wt    & theHit  & 6.528350\\\\\n",
       "\t NrlKO & theBore & 7.477421\\\\\n",
       "\t NrlKO & theHit  & 6.612632\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 4 × 3\n",
       "\n",
       "| gType &lt;fct&gt; | gene &lt;fct&gt; | mean(gExp) &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| wt    | theBore | 7.643900 |\n",
       "| wt    | theHit  | 6.528350 |\n",
       "| NrlKO | theBore | 7.477421 |\n",
       "| NrlKO | theHit  | 6.612632 |\n",
       "\n"
      ],
      "text/plain": [
       "  gType gene    mean(gExp)\n",
       "1 wt    theBore 7.643900  \n",
       "2 wt    theHit  6.528350  \n",
       "3 NrlKO theBore 7.477421  \n",
       "4 NrlKO theHit  6.612632  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "miniDat %>%\n",
    "    group_by(gType,gene) %>% \n",
    "    summarize(mean(gExp))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "lm(formula = gExp ~ devStage, data = miniDat, subset = gene == \n",
       "    \"theHit\")\n",
       "\n",
       "Residuals:\n",
       "     Min       1Q   Median       3Q      Max \n",
       "-0.78538 -0.13299 -0.04788  0.17062  0.51814 \n",
       "\n",
       "Coefficients:\n",
       "                Estimate Std. Error t value Pr(>|t|)    \n",
       "(Intercept)       5.5409     0.1021  54.249  < 2e-16 ***\n",
       "devStageP2        0.3040     0.1399   2.174   0.0368 *  \n",
       "devStageP6        0.2434     0.1399   1.740   0.0909 .  \n",
       "devStageP10       0.8343     0.1399   5.965 9.56e-07 ***\n",
       "devStage4_weeks   3.6325     0.1399  25.973  < 2e-16 ***\n",
       "---\n",
       "Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1\n",
       "\n",
       "Residual standard error: 0.2702 on 34 degrees of freedom\n",
       "Multiple R-squared:  0.9663,\tAdjusted R-squared:  0.9623 \n",
       "F-statistic: 243.4 on 4 and 34 DF,  p-value: < 2.2e-16\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "summary(lm(gExp ~ devStage, miniDat, subset = gene == \"theHit\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Slide 38"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "twoFactFit <- lm(formula = gExp ~ gType * devStage, data = miniDat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in cbind(sampleMeans = theAvgs, minuRef = theAvgs - theAvgs[\"wt.E16\"], :\n",
      "\"number of rows of result is not a multiple of vector length (arg 3)\""
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 2 × 5 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Irs4</th><th scope=col>Nrl</th><th scope=col>Irs4</th><th scope=col>Nrl</th><th scope=col>twoFactFit</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>wt</th><td>7.765750</td><td>11.244200</td><td>NA</td><td>NA</td><td>6.53737500</td></tr>\n",
       "\t<tr><th scope=row>NrlKO</th><td>7.739684</td><td> 6.089632</td><td>NA</td><td>NA</td><td>0.01195833</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 2 × 5 of type dbl\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & Irs4 & Nrl & Irs4 & Nrl & twoFactFit\\\\\n",
       "\\hline\n",
       "\twt & 7.765750 & 11.244200 & NA & NA & 6.53737500\\\\\n",
       "\tNrlKO & 7.739684 &  6.089632 & NA & NA & 0.01195833\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 2 × 5 of type dbl\n",
       "\n",
       "| <!--/--> | Irs4 | Nrl | Irs4 | Nrl | twoFactFit |\n",
       "|---|---|---|---|---|---|\n",
       "| wt | 7.765750 | 11.244200 | NA | NA | 6.53737500 |\n",
       "| NrlKO | 7.739684 |  6.089632 | NA | NA | 0.01195833 |\n",
       "\n"
      ],
      "text/plain": [
       "      Irs4     Nrl       Irs4 Nrl twoFactFit\n",
       "wt    7.765750 11.244200 NA   NA  6.53737500\n",
       "NrlKO 7.739684  6.089632 NA   NA  0.01195833"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cbind(sampleMeans = theAvgs,\n",
    "      minuRef = theAvgs - theAvgs[\"wt.E16\"],\n",
    "      twoFactFit = coef(twoFactFit))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "&lt;NA&gt;"
      ],
      "text/latex": [
       "<NA>"
      ],
      "text/markdown": [
       "&lt;NA&gt;"
      ],
      "text/plain": [
       "[1] NA"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "theAvgs[\"NrlKO.4_weeks\"] -\n",
    "    (theAvgs[\"wt.E16\"] +\n",
    "        (theAvgs[\"NrlKO.E16\"] - theAvgs[\"wt.E16\"]) +\n",
    "             (theAvgs[\"wt.4_weeks\"] - theAvgs[\"wt.E16\"]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "lm(formula = gExp ~ gType * devStage, data = miniDat)\n",
       "\n",
       "Residuals:\n",
       "     Min       1Q   Median       3Q      Max \n",
       "-1.18437 -0.82788 -0.05175  0.80825  1.43067 \n",
       "\n",
       "Coefficients:\n",
       "                           Estimate Std. Error t value Pr(>|t|)    \n",
       "(Intercept)                 6.53737    0.32754  19.959  < 2e-16 ***\n",
       "gTypeNrlKO                  0.01196    0.50032   0.024    0.981    \n",
       "devStageP2                  0.20350    0.46321   0.439    0.662    \n",
       "devStageP6                  0.06163    0.46321   0.133    0.895    \n",
       "devStageP10                 0.45738    0.46321   0.987    0.327    \n",
       "devStage4_weeks             2.02125    0.46321   4.364 4.46e-05 ***\n",
       "gTypeNrlKO:devStageP2      -0.06533    0.68182  -0.096    0.924    \n",
       "gTypeNrlKO:devStageP6       0.12829    0.68182   0.188    0.851    \n",
       "gTypeNrlKO:devStageP10     -0.05533    0.68182  -0.081    0.936    \n",
       "gTypeNrlKO:devStage4_weeks -0.39683    0.68182  -0.582    0.562    \n",
       "---\n",
       "Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1\n",
       "\n",
       "Residual standard error: 0.9264 on 68 degrees of freedom\n",
       "Multiple R-squared:  0.3829,\tAdjusted R-squared:  0.3012 \n",
       "F-statistic: 4.688 on 9 and 68 DF,  p-value: 7.517e-05\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "summary(twoFactFit)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Slide 40"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(predvars, data, env): object 'prMat' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(predvars, data, env): object 'prMat' not found\nTraceback:\n",
      "1. summary(lm(formula = prMat ~ gType * devStage, data = prDes))",
      "2. lm(formula = prMat ~ gType * devStage, data = prDes)",
      "3. eval(mf, parent.frame())",
      "4. eval(mf, parent.frame())",
      "5. stats::model.frame(formula = prMat ~ gType * devStage, data = prDes, \n .     drop.unused.levels = TRUE)",
      "6. model.frame.default(formula = prMat ~ gType * devStage, data = prDes, \n .     drop.unused.levels = TRUE)",
      "7. eval(predvars, data, env)",
      "8. eval(predvars, data, env)"
     ]
    }
   ],
   "source": [
    "summary(lm(formula = prMat ~ gType * devStage, data = prDes))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Slide 41"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary(lm(formula = prMatSimple ~ gType * devStage, data = prDes))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
