<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
  <head>
    <title>Statistical Methods for  High Dimensional Biology STAT/BIOF/GSAT 540</title>
    <meta charset="utf-8" />
    <link href="libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link href="libs/remark-css-0.0.1/default-fonts.css" rel="stylesheet" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Statistical Methods for <br>High Dimensional Biology<br><font size=8>STAT/BIOF/GSAT 540</font>
## <font color=red> Lecture 7 â€“ Linear Models</font> <br style=line-height:2> Gabriela Cohen Freue
### <font size=5> Other contributors: Jenny Bryan, Sara Mostafavi, Su-In Lee</font>

---

class: middle

# Recall from last class...
---

class: center, middle

![](L7_LinearModels_files/ttest_to_reg.png)
---
class: center

![](L7_LinearModels_files/LM_vbles.png)
---
class: center

![](L7_LinearModels_files/param_dummy.png)
---
class: middle
### Different ways of writing this (design matrix, parameter vector) pair correspond to different parametrizations of the model.
# `$$Y = [X\alpha] + \varepsilon$$`
&lt;br&gt; &lt;big&gt;
 Understanding these concepts makes it easier ...
- to interpret fitted models with confidence
- to fit models such that comparisons you care most about are directly addressed in the inferential "report"
---
class: middle

### Recall we can obtain one set of parameters from the others!
&lt;center&gt;
&lt;img src="L7_LinearModels_files/diff_means.png" height="250px" /&gt;
&lt;/center&gt;
By default, `lm` estimates these parameters!


```r
summary(lm(gExp~devStage,subset(devDat,gene=="theHit")))$coeff
```

```
##                  Estimate Std. Error   t value     Pr(&gt;|t|)
## (Intercept)     5.5408571  0.1021381 54.248698 1.307554e-34
## devStageP2      0.3040179  0.1398583  2.173756 3.678022e-02
## devStageP6      0.2433929  0.1398583  1.740282 9.085489e-02
## devStageP10     0.8342679  0.1398583  5.965093 9.559065e-07
## devStage4_weeks 3.6325179  0.1398583 25.972843 5.266481e-24
```

---
class: middle

```r
#summary(hitFit)
```
---
class: middle
### **Differential Expression**: Are the expression levels of these genes are generated by distributions with different locations at different developmental stages? 

### `$$H_0: \mu_{E12}=\mu_{P2}=\mu_{P6}=\mu_{P10}=\mu_{4W}$$`



![](L7_LinearModels_files/figure-html/unnamed-chunk-5-1.svg)&lt;!-- --&gt;

---
class: middle
#NEW HERE
---
class: middle
## Increasing the complexity of the linear model ...

### What if you have two categorical variables?
&lt;Big&gt;
  
  e.g., `gType` and `devStage` &lt;font size=3&gt;
    (for simplicity, let's consider only E16 and 4W)&lt;/font&gt;

&gt; Is the assocation between gene expression and genotype different at different developmental stages?

![](L6_ANOVA_files/slide32.png)


---
class: middle


## Two-way ANOVA or regression with interaction 

### What parameters are we estimating? 

### What hypotheses are we testing?


```r
twoFactFit &lt;- lm(gExp ~ gType * devStage, twoDat)
summary(twoFactFit)$coeff
```

```
##                              Estimate Std. Error   t value     Pr(&gt;|t|)
## (Intercept)                 9.9080000  0.1574912 62.911469 2.027211e-15
## gTypeNrlKO                 -0.9856667  0.2405717 -4.097184 1.767824e-03
## devStage4_weeks             1.6345000  0.2227261  7.338609 1.469261e-05
## gTypeNrlKO:devStage4_weeks -2.0380833  0.3278440 -6.216626 6.560671e-05
```
---
class: middle
![](L6_ANOVA_files/slide33.png)


```
##                              Estimate Std. Error   t value     Pr(&gt;|t|)
*## (Intercept)                 9.9080000  0.1574912 62.911469 2.027211e-15
## gTypeNrlKO                 -0.9856667  0.2405717 -4.097184 1.767824e-03
## devStage4_weeks             1.6345000  0.2227261  7.338609 1.469261e-05
## gTypeNrlKO:devStage4_weeks -2.0380833  0.3278440 -6.216626 6.560671e-05
```
---
class: middle
$$ Y_{ijk}=\theta+\tau_i+\tau_{B}$$

model parameter | R | stats
--------|---------|---------
`\(\theta\)` | (Intercept) | `\(E[Y_{wt,E16}]\)`
`\(\tau_A\)` | gTypeNrlKO | *conditional* effect of NrlKO *at* E16
`\(\tau_B\)` | devStage4_weeks | *conditional* effect of 4_weeks *at* WT
`\(\tau_{AB}\)` | gTypeNrlKO:devStage4_weeks | *interaction* effect of NrlKO and 4_weeks
---
class: middle
![](L6_ANOVA_files/slide34.png)


```
##                              Estimate Std. Error   t value     Pr(&gt;|t|)
## (Intercept)                 9.9080000  0.1574912 62.911469 2.027211e-15
*## gTypeNrlKO                 -0.9856667  0.2405717 -4.097184 1.767824e-03
## devStage4_weeks             1.6345000  0.2227261  7.338609 1.469261e-05
## gTypeNrlKO:devStage4_weeks -2.0380833  0.3278440 -6.216626 6.560671e-05
```
---
class: middle
![](L6_ANOVA_files/slide35.png)


```
##                              Estimate Std. Error   t value     Pr(&gt;|t|)
## (Intercept)                 9.9080000  0.1574912 62.911469 2.027211e-15
## gTypeNrlKO                 -0.9856667  0.2405717 -4.097184 1.767824e-03
*## devStage4_weeks             1.6345000  0.2227261  7.338609 1.469261e-05
## gTypeNrlKO:devStage4_weeks -2.0380833  0.3278440 -6.216626 6.560671e-05
```

---
class: middle
![](L6_ANOVA_files/slide36.png)


```
##                              Estimate Std. Error   t value     Pr(&gt;|t|)
## (Intercept)                 9.9080000  0.1574912 62.911469 2.027211e-15
## gTypeNrlKO                 -0.9856667  0.2405717 -4.097184 1.767824e-03
## devStage4_weeks             1.6345000  0.2227261  7.338609 1.469261e-05
*## gTypeNrlKO:devStage4_weeks -2.0380833  0.3278440 -6.216626 6.560671e-05
```

---
class: middle
![](L6_ANOVA_files/slide37.png)
---
class: middle
![](L6_ANOVA_files/slide38.png)
---
class: middle

## Let's go through some example genes to get a sense of what an interaction effect looks like.
     
     We have three parameters we'd like to interpret:
.pull-left[
&gt; ### Main effect: genotype
### Main effect: age
### Interaction: genotype*age
]
.pull-right[
&lt;br&gt;
![](L6_ANOVA_files/slide39.png)
]

---
class: middle
![](L6_ANOVA_files/slide40.png)
---
class: middle
![](L6_ANOVA_files/slide41.png)
---
class: middle
![](L6_ANOVA_files/slide42.png)
---
class: middle
![](L6_ANOVA_files/slide43.png)
---
class: middle
![](L6_ANOVA_files/slide44.png)
---
class: center

![](L7_LinearModels_files/slide12.png)
---
class: middle




```
## 
## Call:
## lm(formula = gExp ~ gType * devStage, data = miniDat)
## 
## Residuals:
##     Min      1Q  Median      3Q     Max 
## -2.4521 -1.3117  0.3066  1.1132  2.3268 
## 
## Coefficients:
##                            Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)                  8.6101     0.5607  15.356   &lt;2e-16 ***
## gTypeNrlKO                  -1.7878     0.8565  -2.087   0.0406 *  
## devStageP2                   0.6915     0.7929   0.872   0.3862    
## devStageP6                   1.1386     0.7929   1.436   0.1556    
## devStageP10                  1.2431     0.7929   1.568   0.1216    
## devStage4_weeks              1.4010     0.7929   1.767   0.0817 .  
## gTypeNrlKO:devStageP2       -0.5971     1.1672  -0.512   0.6106    
## gTypeNrlKO:devStageP6       -1.0648     1.1672  -0.912   0.3648    
## gTypeNrlKO:devStageP10      -1.2592     1.1672  -1.079   0.2845    
## gTypeNrlKO:devStage4_weeks  -1.1146     1.1672  -0.955   0.3430    
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## Residual standard error: 1.586 on 68 degrees of freedom
## Multiple R-squared:  0.4527,	Adjusted R-squared:  0.3802 
## F-statistic: 6.249 on 9 and 68 DF,  p-value: 2.192e-06
```
---
class: center

![](L7_LinearModels_files/slide13.png)
---
class: center

![](L7_LinearModels_files/slide14.png)
---
class: middle
## **How** do we test the **overall** association between the response and a factor?
&lt;big&gt;
Is it easier to use an additive model?
- Additive models are easier and smaller (fewer parameters)
- But in some applications, we need to test the interaction term.
- And it does not always test the overall association between a factor and a response.

![](L7_LinearModels_files/slide15.png)

---
class: middle
## **How** do we test the **overall** association between the response and a factor?

![](L7_LinearModels_files/slide16.png)
---
class: middle
## **How** do we test the **overall** association between the response and a factor?

![](L7_LinearModels_files/slide17.png)
---
class: middle

```r
summary(lm(formula = gExp ~ gType +devStage, data = miniDat))
```

```
## 
## Call:
## lm(formula = gExp ~ gType + devStage, data = miniDat)
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -2.31027 -1.45530 -0.02388  1.16338  2.54091 
## 
## Coefficients:
##                 Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)       8.9662     0.4432  20.229  &lt; 2e-16 ***
## gTypeNrlKO       -2.6187     0.3536  -7.406 1.96e-10 ***
## devStageP2        0.4523     0.5709   0.792    0.431    
## devStageP6        0.6656     0.5709   1.166    0.248    
## devStageP10       0.6729     0.5709   1.179    0.242    
## devStage4_weeks   0.9031     0.5709   1.582    0.118    
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## Residual standard error: 1.558 on 72 degrees of freedom
## Multiple R-squared:  0.4403,	Adjusted R-squared:  0.4014 
## F-statistic: 11.33 on 5 and 72 DF,  p-value: 4.436e-08
```
---
Slide 19

---
Slide 20

---
class: middle
## Summary so far
&lt;big&gt;
- *t*-tests can be used to test the equality of **2** population means.

- ANOVA can be used to test the equality of **more than 2** population means.

- **Linear regression** provides a general framework for modelling the relationship between response variable and different type of explanatory variables.

- ***t*-tests** can be used to test the significance of *individual* coefficients.

- **F-tests** can be used to test the *simultaneous significance of multiple coefficients*. We need it to test the association between a response and categorical variable.
---
class: center

![](L7_LinearModels_files/slide22.png)
---

SLIDE 23

---

slide 24

---
class: center

![](L7_LinearModels_files/slide25.png)
---
class: center

![](L7_LinearModels_files/slide26.png)
---

SLIDE 27

---

SLIDE 28

---
class: center, middle

## How do we estimate the intercept and the slope?

### Is there an optimal line?

---
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"highlightSpans": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
