<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
  <head>
    <title> Lecture 7 – Linear Models</title>
    <meta charset="utf-8" />
    <meta name="author" content="Keegan Korthauer" />
    <link href="libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link href="libs/remark-css-0.0.1/default-fonts.css" rel="stylesheet" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# <font color=red> Lecture 7 – Linear Models</font>
## STAT/BIOF/GSAT 540: Statistical Methods for High Dimensional Biology
### Keegan Korthauer
### 2020/01/27 <br><br><font size=5> Slides by: Gabriela Cohen Freue with contributions from Jenny Bryan and Keegan Korthauer </font>

---

class: middle



# Recall from last class...

### - show how to compare means of different groups (2 or more) using a linear regression model

  - dummy variables to model the levels of a qualitative explanatory variable

### - write a linear model using matrix notation

  - understand which matrix is built by R
  
### - distinguish between single and multiple hypotheses 
  - `\(t\)`-tests vs `\(F\)`-tests  
---
class: middle

## **Quick review**: from `\(t\)`-test to linear regression
![](L7_LinearModels_files/ttest_to_reg.png)
---
class: middle

## &lt;font color="blue"&gt; **HOW??** &lt;/font&gt;
### Changing the parametrization and using dummy variables

&lt;big&gt;
$$
Y \sim G; \ E[Y]= \mu_Y ; \ Z \sim G; \ E[Z]= \mu_Z
$$

&lt;center&gt;**↓**

`$$Y_{ij}=\theta+\tau_2 \times x_{ij2} + \varepsilon_{ij}; \ i=1, \dots, n; \ j={1,2}$$`

&lt;center&gt;**↓**


.pull-left[
&lt;br&gt;
&lt;br&gt;
&lt;br&gt;
`$$\begin{array}{l}
E[Y_{i1}] &amp;= \theta=\mu_1 \\
E[Y_{i2}] &amp;= \theta + \tau_2=\mu_1\ + (\mu_2 - \mu_1) = \mu_2
\end{array}$$`

]

.pull-right[
&lt;img src="L7_LinearModels_files/param.png" width="232" height="150px" /&gt;
]
---
class: middle

## **Using matrix notation ...**

&lt;img src="L7_LinearModels_files/matrix_not.png" width="2447" height="400px" /&gt;

### ... and similarly beyond 2 groups comparisons (ANOVA)

---
class: center

![](L7_LinearModels_files/LM_vbles.png)
---
class: middle
##**Parametrizations**

### Different ways of writing this [design matrix, parameter vector] pair correspond to different parametrizations of the model
# `$$Y = [X\alpha] + \varepsilon$$`
&lt;br&gt; &lt;big&gt;
 Understanding these concepts makes it easier ...
- to interpret fitted models
- to fit models such that comparisons you care most about are directly addressed in the inferential "report"
---
class: middle

### For example: comparisons of mean expression levels between groups!
&lt;center&gt;
&lt;img src="L7_LinearModels_files/diff_means.png" width="1908" height="250px" /&gt;
&lt;/center&gt;
By default, `lm` estimates mean differences (with respect to a reference group):


```r
summary(lm(gExp~devStage,subset(devDat,gene=="theHit")))$coeff
```

```
##              Estimate Std. Error   t value     Pr(&gt;|t|)
## (Intercept) 5.5408571  0.1021381 54.248698 1.307554e-34
## devStageP2  0.3040179  0.1398583  2.173756 3.678022e-02
## devStageP6  0.2433929  0.1398583  1.740282 9.085489e-02
## devStageP10 0.8342679  0.1398583  5.965093 9.559065e-07
## devStage4W  3.6325179  0.1398583 25.972843 5.266481e-24
```
---
class: middle
## Today... more complex models

### - more than one factor with multiple levels

  - how to model many categorical variables and their interaction

### - distinguish between simple and main effects 
  - `lm` vs `anova` tests
  
### - nested models 
  - `\(t\)`-tests vs `\(F\)`-tests  
  
### - continuous explanatory variables

  - the regression line
  

---
class: middle

## Increasing the complexity of the linear model ...

### What if you have two categorical variables?

  
  e.g., `gType` and `devStage` (for simplicity, let's consider only E16 and 4W)

- ANOVA is usually used to study models with one or more categorical variables (factors)

- Can we combine levels into 4 groups to simplify the analysis??


![](L7_LinearModels_files/anova.png)


---
class: middle


## Two-way ANOVA or a linear model with interaction 

Which group means are we comparing in a model with 2 factors? 

![](L7_LinearModels_files/figure-html/unnamed-chunk-7-1.svg)&lt;!-- --&gt;

`$$\mu_1=E[Y_{(wt,E16)}], \ \mu_2=E[Y_{(NrlKO,E16)}], 
\ \mu_3=E[Y_{(wt,4W)}], \ \mu_4=E[Y_{(NrlKO,4W)}]$$`

---
class: middle
### Reference-treatment effect parametrization

By default, `lm` assumes a &lt;font color = "red"&gt;**reference-treatment effect**&lt;/font&gt; parametrization (mathematically, we need *more* dummy variables, see [math handout](../SupportingMaterial/math_handout.html))


```r
twoFactFit &lt;- lm(gExp ~ gType * devStage, twoDat)
```


```
##                         Estimate Std. Error   t value     Pr(&gt;|t|)
## (Intercept)            9.9080000  0.1574912 62.911469 2.027211e-15
## gTypeNrlKO            -0.9856667  0.2405717 -4.097184 1.767824e-03
## devStage4W             1.6345000  0.2227261  7.338609 1.469261e-05
## gTypeNrlKO:devStage4W -2.0380833  0.3278440 -6.216626 6.560671e-05
```


```r
means.2Fact &lt;- as.data.frame(twoDat %&gt;% 
          group_by(grp) %&gt;% summarize(cellMeans=mean(gExp)))
(means.2Fact &lt;-means.2Fact %&gt;%
          mutate(txEffects=cellMeans-cellMeans[1],
          lmEst=summary(twoFactFit)$coeff[,1]))
```

```
##         grp cellMeans  txEffects      lmEst
## 1    wt.E16  9.908000  0.0000000  9.9080000
## 2 NrlKO.E16  8.922333 -0.9856667 -0.9856667
## 3     wt.4W 11.542500  1.6345000  1.6345000
## 4  NrlKO.4W  8.518750 -1.3892500 -2.0380833
```

---
class: middle

## The reference: &lt;font color="blue"&gt; wt &amp; E16 &lt;/font&gt;

&lt;big&gt;
As before, comparisons are relative to a reference but in this case there is a reference level in each factor: &lt;font color="blue"&gt;wt and E16 &lt;/font&gt;

![](L7_LinearModels_files/figure-html/unnamed-chunk-11-1.svg)&lt;!-- --&gt;

---
class: middle

## The reference: wt &amp; E16
&lt;big&gt;

**Mean of reference group**: `\(\theta=E[Y_{wt,E16}]\)` 
  
**`lm` estimate**: `\(\hat{\theta}\)` is the sample mean of the group 


```
##                         Estimate Std. Error   t value     Pr(&gt;|t|)
*## (Intercept)            9.9080000  0.1574912 62.911469 2.027211e-15
## gTypeNrlKO            -0.9856667  0.2405717 -4.097184 1.767824e-03
## devStage4W             1.6345000  0.2227261  7.338609 1.469261e-05
## gTypeNrlKO:devStage4W -2.0380833  0.3278440 -6.216626 6.560671e-05
```

```
##         grp cellMeans  txEffects      lmEst
*## 1    wt.E16  9.908000  0.0000000  9.9080000
## 2 NrlKO.E16  8.922333 -0.9856667 -0.9856667
## 3     wt.4W 11.542500  1.6345000  1.6345000
## 4  NrlKO.4W  8.518750 -1.3892500 -2.0380833
```

In general, one is not interested in: `\(H_0: \theta=0\)`

---
class: middle

## Simple genotype effect: wt *vs* NrlKO &lt;font color="blue"&gt; at E16 &lt;/font&gt;

&lt;big&gt;
And now the "treatment effects"... 

&gt; **Important**: effects are not marginal but *conditional* effects (&lt;font color="blue"&gt;at a given level &lt;/font&gt; of the other factor, e.g., at E16), usually called **simple effects**

![](L7_LinearModels_files/figure-html/unnamed-chunk-13-1.svg)&lt;!-- --&gt;

---

class: middle

## Simple genotype effect: wt *vs* NrlKO &lt;font color="blue"&gt; at E16 &lt;/font&gt;

**Effect of genotype at E16**: `\(\tau_{KO}=E[Y_{NrlKO,E16}]-E[Y_{wt,E16}]\)`

**`lm` estimate**: `\(\hat{\tau}_{KO}\)` is the *difference* of sample respective means (check below)


```
##                         Estimate Std. Error   t value     Pr(&gt;|t|)
## (Intercept)            9.9080000  0.1574912 62.911469 2.027211e-15
*## gTypeNrlKO            -0.9856667  0.2405717 -4.097184 1.767824e-03
## devStage4W             1.6345000  0.2227261  7.338609 1.469261e-05
## gTypeNrlKO:devStage4W -2.0380833  0.3278440 -6.216626 6.560671e-05
```

```
##         grp cellMeans  txEffects      lmEst
## 1    wt.E16  9.908000  0.0000000  9.9080000
*## 2 NrlKO.E16  8.922333 -0.9856667 -0.9856667
## 3     wt.4W 11.542500  1.6345000  1.6345000
## 4  NrlKO.4W  8.518750 -1.3892500 -2.0380833
```

**But**, do you want to test the *conditional* effect at E16: `\(H_0: \tau_{KO}=0\)`?? 

---

class: middle
### Simple developmental effect: E16 *vs* 4W &lt;font color="blue"&gt; at wt &lt;/font&gt;

&lt;big&gt;

Similarly, for the other factor:

![](L7_LinearModels_files/figure-html/unnamed-chunk-15-1.svg)&lt;!-- --&gt;

---

class: middle
### Simple developmental effect: E16 *vs* 4W &lt;font color="blue"&gt; at wt &lt;/font&gt;

**Effect of development at wt**: `\(\tau_{4W}=E[Y_{wt,4W}]-E[Y_{wt,E16}]\)`

**`lm` estimate**: `\(\hat{\tau}_{4W}\)` is the *difference* of respective sample means (check below)


```
##                         Estimate Std. Error   t value     Pr(&gt;|t|)
## (Intercept)            9.9080000  0.1574912 62.911469 2.027211e-15
## gTypeNrlKO            -0.9856667  0.2405717 -4.097184 1.767824e-03
*## devStage4W             1.6345000  0.2227261  7.338609 1.469261e-05
## gTypeNrlKO:devStage4W -2.0380833  0.3278440 -6.216626 6.560671e-05
```

```
##         grp cellMeans  txEffects      lmEst
## 1    wt.E16  9.908000  0.0000000  9.9080000
## 2 NrlKO.E16  8.922333 -0.9856667 -0.9856667
*## 3     wt.4W 11.542500  1.6345000  1.6345000
## 4  NrlKO.4W  8.518750 -1.3892500 -2.0380833
```

---
class: middle

## Interaction effect

&lt;big&gt;

Is the effect of genotype the same at different developmental stages? (or does hte effect of development depend on genotype?) 

**Yes if**, there's no interaction effect, i.e., `\(\tau_{KO4W}=0\)` 

![](L7_LinearModels_files/figure-html/unnamed-chunk-17-1.svg)&lt;!-- --&gt;

The genotype effect at E16 is `\(\tau_{KO}\)`. However, `\(\tau_{KO}\)` does not seem to be the effect at 4W. The difference is the interaction effect!
---
## Interaction effect

`\(\tau_{KO4W}=(E[Y_{NrlKO,4W}]-E[Y_{wt,4W}])-(E[Y_{NrlKO,E16}]-E[Y_{wt,E16}])\)` 


```
##                         Estimate Std. Error   t value     Pr(&gt;|t|)
## (Intercept)            9.9080000  0.1574912 62.911469 2.027211e-15
## gTypeNrlKO            -0.9856667  0.2405717 -4.097184 1.767824e-03
## devStage4W             1.6345000  0.2227261  7.338609 1.469261e-05
*## gTypeNrlKO:devStage4W -2.0380833  0.3278440 -6.216626 6.560671e-05
```


```r
means.2Fact
```

```
##         grp cellMeans  txEffects      lmEst
## 1    wt.E16  9.908000  0.0000000  9.9080000
## 2 NrlKO.E16  8.922333 -0.9856667 -0.9856667
## 3     wt.4W 11.542500  1.6345000  1.6345000
*## 4  NrlKO.4W  8.518750 -1.3892500 -2.0380833
```

```r
((means.2Fact$cellMeans[4]-means.2Fact$cellMeans[3])-
    (means.2Fact$cellMeans[2]-means.2Fact$cellMeans[1]))
```

```
## [1] -2.038083

NA
NA
*NA```

---
class: middle
## Summary of model parameters: with interaction 

model parameter | R estimate| stats | interpretation
--------|---------|---------
`\(\theta\)` | (Intercept) | `\(E[Y_{wt,E16}]\)` | reference
`\(\tau_{KO}\)` | gTypeNrlKO | `\(E[Y_{NrlKO,E16}] - E[Y_{wt,E16}]\)` | *conditional* effect of NrlKO *at* E16
`\(\tau_{4W}\)` | devStage4_weeks | `\(E[Y_{wt,4W}] - E[Y_{wt,E16}]\)` | *conditional* effect of 4W *at* wt
`\(\tau_{KO4W}\)` | gTypeNrlKO: devStage4_weeks | `\(E[Y_{NrlKO,4W}] - E[Y_{wt,4W}] - \tau_{KO}\)`  | *interaction* effect of NrlKO and 4W

It is *important* to remember that `lm` reports *simple, not main* effects!! &lt;font color="blue"&gt; why?? because of the parametrization used!! &lt;/font&gt; (see see [math handout](../SupportingMaterial/math_handout.html))

It can also be shown that `\(\tau_{KO4W}=E[Y_{NrlKO,4W}]-\tau_{4W}-\tau_{KO}-\theta\)` 

---
class: middle

### Let's examine these parameters closer and some examples

&lt;big&gt;     
For our model, `lm` tests 4 hypotheses:

.pull-left[
&gt; ### `\(H_0: \theta=0\)`
### `\(H_0: \tau_{KO}=0\)`
### `\(H_0: \tau_{4W}=0\)`
### `\(H_0: \tau_{KO4W}=0\)`
]
.pull-right[
&lt;br&gt;
![](L7_LinearModels_files/2way_param.png)
]

We may not be interested in these hypotheses, e.g., `\(\tau_{KO}\)` and `\(\tau_{4W}\)` are *conditional* effects *at* a given level of a factor (*simple effects*)
---
class: middle

### Example 1: nothing is statistically significant, very flat genes `\(^*\)`

```r
egDat&lt;-prDat %&gt;% subset(row.names(prDat) %in%
       c("1442080_at","1448243_at")) %&gt;%
       tibble::rownames_to_column(var = "gene")%&gt;% 
       gather(sidChar, gExp,-gene) %&gt;%
       inner_join(prDes,by="sidChar")
```
&lt;small&gt; `\(^*\)` Here and in next slides, summary of `lm` shown for the gene in the left plot


```
##                         Estimate Std. Error    t value     Pr(&gt;|t|)
## (Intercept)            8.5240000  0.2561305 33.2799114 2.154028e-12
*## gTypeNrlKO            -0.4336667  0.3912458 -1.1084251 2.913251e-01
*## devStage4W            -0.2532500  0.3622232 -0.6991545 4.989723e-01
*## gTypeNrlKO:devStage4W  0.5504167  0.5331781  1.0323317 3.240804e-01
```

![](L7_LinearModels_files/figure-html/unnamed-chunk-22-1.svg)&lt;!-- --&gt;

---
class: middle
### Example 2: statistically significant interaction effect: non-parallel 





```
##                       Estimate Std. Error   t value     Pr(&gt;|t|)
## (Intercept)            8.58550  0.2215161 38.757897 4.083165e-13
*## gTypeNrlKO            -1.59250  0.3383715 -4.706366 6.434783e-04
*## devStage4W            -2.22075  0.3132711 -7.088907 2.021975e-05
*## gTypeNrlKO:devStage4W  4.96975  0.4611226 10.777502 3.481389e-07
```

![](L7_LinearModels_files/figure-html/unnamed-chunk-25-1.svg)&lt;!-- --&gt;

When the interaction effect is significant, the *simple* effects may not agree: compare the genotype effect @E16 with that @4W!

*Main* effects (overall): does genotype have an effect on gene expression? we can't answer this question! &lt;font color="blue"&gt; it depends (on the level of devStage)! &lt;/font&gt; (more later) 

---
class: middle

### Example 3: balance &amp; only genotype @E16 is statistically significant

To simplify future explanations, I've added a random observation in the NrlKO.E16 group (close to its mean) to have a *balanced* design 
&gt; In *unbalanced* designs the *main* effects are a *weighted* average of the simple effects, and the weights are not easy to interpret (beyond the scope of this course but worth noting the issue!)


```r
egDat&lt;-prDat %&gt;% subset(row.names(prDat) %in%
       c("1447753_at","1431651_at")) %&gt;%
       tibble::rownames_to_column(var = "gene")%&gt;% 
       gather(sidChar, gExp,-gene) %&gt;%
       inner_join(prDes,by="sidChar") %&gt;% 
       mutate(grp=interaction(gType, devStage))

#duplicate sample 6 and add noise to gExp of genes
set.seed(123)
egDat &lt;-egDat %&gt;% subset(sidChar=="Sample_6") %&gt;%
       mutate(sidChar="Sample_r",sidNum="r",
       gExp=gExp+rnorm(2,0,.1)) %&gt;% rbind(egDat) %&gt;% 
       arrange(grp)
```
---
class: middle

### Example 3: only genotype @E16 is statistically significant: parallel

The interaction effect is not significant (almost parallel pattern).

Thus, there may be a genotype effect *regardless* of the developmental stage (*main* effect). However, that hypothesis is *not* tested here!!  

How do we test a *main effect*??  


```
##                          Estimate Std. Error    t value     Pr(&gt;|t|)
## (Intercept)            6.47725000 0.04513333 143.513681 8.795702e-21
*## gTypeNrlKO             0.28699556 0.06382816   4.496378 7.312501e-04
## devStage4W            -0.07675000 0.06382816  -1.202447 2.523805e-01
## gTypeNrlKO:devStage4W -0.09949556 0.09026666  -1.102240 2.919732e-01
```

![](L7_LinearModels_files/figure-html/unnamed-chunk-28-1.svg)&lt;!-- --&gt;

---
class: middle
## **How** do we test the *main* effects?

The main effect measures the *overall* association between the response and a factor. They are the (weighted) average of an effect over the levels of the other factor

&gt; **Note**: a significant interaction means that the effect of a factor depends on the level of the other one. Thus, main effects may mask interesting results!


![](L7_LinearModels_files/figure-html/unnamed-chunk-29-1.svg)&lt;!-- --&gt;

---
class: middle

## Main effects

`anova()` can be used to test the main effects:

`$$H_0: ((E[Y_{KO,E16}]-E[Y_{wt,E16}])+(E[Y_{KO,4W}]-E[Y_{wt,4W}]))/2=0$$`
&gt; for unbalanced experiments `\(H_0: w_1 \text{effect}_{E16}+ w_2 \text{effect}_{4W}=0\)` 


```r
tidy(anova(lm(gExp ~ gType * devStage, plot1Dat)))
```

```
## # A tibble: 4 x 6
##   term              df   sumsq  meansq statistic   p.value
##   &lt;chr&gt;          &lt;int&gt;   &lt;dbl&gt;   &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;
*## 1 gType              1 0.225   0.225       27.6   0.000202
*## 2 devStage           1 0.0640  0.0640       7.86  0.0160  
## 3 gType:devStage     1 0.00990 0.00990      1.21  0.292   
## 4 Residuals         12 0.0978  0.00815     NA    NA
```

As we suspected in slide #26, there is a significant genotype effect for this gene (1447753_at), i.e., its mean expression changes in NrlKO group (compared to wt), on average over develpmental stages.

&lt;small&gt;
`\(^*\)` `anova` uses type II sums of squares, which follows the principle of marginality, thus order matters in unbalanced designs!
---
# Main &amp; interaction effects: important notes

&lt;Big&gt;

- A **significant interaction effect** means that the effect of one factor depends on the levels of the other one. 
  - e.g., the effect of genotype depends on development

- **Main effects**: are the (weighted) average of an effect over the levels of the other factor. 

- A **non-significant main effect** means that, on average, there's no evidence of a factor's effect
  - e.g, no evidence of a genotype effect, on average over both developmental stages

- **Note of caution**: if the interaction is significant, it is possible that one or both simple effects are significant but the average effect (i.e., the main effect) is not. This is because the effect of a factor *depends on* the level of the other one!

---
# Additive models

&lt;Big&gt;

- In some applications, we need to test the interaction term

- However, additive models are easier and smaller

- If there are no statistical or theoretical grounds to include the interaction term, additive models are preferred 

- Additive effects: `\(E[Y_{NrlKO,4W}]-E[Y_{wt,E16}]=\tau_{KO}+\tau_{4W}\)`


```r
addFit &lt;- summary(lm(formula = gExp ~ gType +devStage,plot1Dat))$coeff
addFit
```

```
##               Estimate Std. Error    t value     Pr(&gt;|t|)
## (Intercept)  6.5021239  0.0394084 164.993346 5.609641e-23
## gTypeNrlKO   0.2372478  0.0455049   5.213675 1.670701e-04
## devStage4W  -0.1264978  0.0455049  -2.779872 1.561988e-02
```

&lt;small&gt;
`\(^*\)` gene=1447753_at in table

---
&lt;big&gt;
**Additive models**

- In an additive model, the parameters are **average effects**, over the levels of the other factor. Now, same as in anova()!!

&gt; Note the agreement!! This is gone in unbalanced designs since weights are computed differently! &lt;font color=blue&gt; try!!&lt;/font&gt;

- TypeIII sum of squares are required for agreement in unbalanced designs (use `Anova` in `car`), beyond our scope



```
##               Estimate Std. Error    t value     Pr(&gt;|t|)
## (Intercept)  6.5021239  0.0394084 164.993346 5.609641e-23
*## gTypeNrlKO   0.2372478  0.0455049   5.213675 1.670701e-04
## devStage4W  -0.1264978  0.0455049  -2.779872 1.561988e-02
```


```r
tidy(anova(lm(gExp ~ gType + devStage,plot1Dat)))
```

```
## # A tibble: 3 x 6
##   term         df  sumsq  meansq statistic   p.value
##   &lt;chr&gt;     &lt;int&gt;  &lt;dbl&gt;   &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;
*## 1 gType         1 0.225  0.225       27.2   0.000167
## 2 devStage      1 0.0640 0.0640       7.73  0.0156  
## 3 Residuals    13 0.108  0.00828     NA    NA
```
---

![](L7_LinearModels_files/figure-html/unnamed-chunk-34-1.svg)&lt;!-- --&gt;


```r
multEst
```

```
##           (Intercept)            gTypeNrlKO            devStage4W 
##            6.47725000            0.28699556           -0.07675000 
## gTypeNrlKO:devStage4W 
##           -0.09949556
```

```r
addEst 
```

```
## (Intercept)  gTypeNrlKO  devStage4W 
##   6.5021239   0.2372478  -0.1264978
```
---
# Factors with multiple levels
&lt;big&gt;

We can generalize the regression model to factors with more levels (e.g., E16, P2, P10 and 4W): we just additional dummy variables (and parameters)!!

&gt; With interaction


```
##                           Estimate Std. Error    t value     Pr(&gt;|t|)
## (Intercept)             5.43325000  0.1240289 43.8063081 4.740219e-28
## gTypeNrlKO              0.25108333  0.1894573  1.3252764 1.954265e-01
## devStageP2              0.39900000  0.1754034  2.2747562 3.049627e-02
## devStageP6              0.19525000  0.1754034  1.1131483 2.747868e-01
## devStageP10             0.92000000  0.1754034  5.2450520 1.283680e-05
## devStage4W              3.96125000  0.1754034 22.5836544 5.952464e-20
## gTypeNrlKO:devStageP2  -0.22583333  0.2581868 -0.8746896 3.889296e-01
## gTypeNrlKO:devStageP6   0.06041667  0.2581868  0.2340037 8.166263e-01
## gTypeNrlKO:devStageP10 -0.20733333  0.2581868 -0.8030361 4.284868e-01
## gTypeNrlKO:devStage4W  -0.69333333  0.2581868 -2.6853939 1.185648e-02
```

Note that all the `devStage` parameters are still *simple* effects, but we now have more: one for each level compared to the reference

---
class: middle
# Factors with multiple levels (cont.)
&lt;big&gt;

&gt; Without interaction: additive


```
##               Estimate Std. Error    t value     Pr(&gt;|t|)
## (Intercept) 5.52731618 0.11010606 50.1999257 9.574287e-33
## gTypeNrlKO  0.03159559 0.08783425  0.3597183 7.213497e-01
## devStageP2  0.30176103 0.14182289  2.1277315 4.091897e-02
## devStageP6  0.24113603 0.14182289  1.7002617 9.848949e-02
## devStageP10 0.83201103 0.14182289  5.8665498 1.428982e-06
## devStage4W  3.63026103 0.14182289 25.5971450 2.412597e-23
```

Parameters are now *main* effects (on average over the levels of the other factor) but we have more!

&lt;font color="blue"&gt; Is developmental a significant effect? &lt;/font&gt; We haven't tested that!!
---
class: middle

# Simultaneous hypotheses again

### We generally test two types of null hypotheses:
      
.pull-left[
&lt;center&gt;
`$$H_0: \tau_j = 0$$`
vs
`$$H_0: \tau_j \neq 0$$`
for each *j* &lt;font color="red"&gt;individually&lt;/font&gt;
          
e.g., Is gene A differencially expressed 2 days after birth?
`$$H_0: \tau_{P2}=0$$`
          ]
    
.pull-right[
&lt;center&gt;

`$$H_0: \tau_j = 0$$`
        vs
`$$H_0: \tau_j \neq 0$$`
for all *j* &lt;font color="red"&gt;at the same time&lt;/font&gt;
        
e.g., Is gene A significantly affected by time (`devStage`)?
        
`$$H_0: \tau_{P2}=\tau_{P6}=\tau_{P10}=\tau_{4W}=0$$`
        ]
---
class: middle
    
## *F*-test and overall significance: a deja vu 
&lt;big&gt;
  - the *t*-test in linear regression allows us to test single hypotheses. Those are given in the summary of `lm`
      `$$H_0 : \tau_i = 0$$`
      `$$H_A : \tau_j \neq 0$$`
  - but we often like to test multiple hypotheses *simultaneously*: 
      `$$H_0 : \tau_{P2} = \tau_{P6} = \tau_{P10} = \tau_{4W}=0\textrm{ [AND statement]}$$`
      `$$H_A : \tau_i \neq 0 \textrm{ for some i [OR statement]}$$` the *F*-test allows us to test such compound tests
      
---
class: middle
# Overall effects: compound tests

### With interaction

&lt;Big&gt;
&gt; `\(H_0: \tau_{KO}=0\)` (1 df) 
`\(H_0: \tau_{P2}=\tau_{P6}=\tau_{P10}=\tau_{4W}=0\)` (at wt!, 4 df)
`\(H_0: \tau_{KOP2}=\tau_{KOP6}=\tau_{KOP10}=\tau_{KO4W}=0\)` (4 df)


```r
tidy(anova(lm(gExp~gType*devStage,hitDat)))
```

```
## # A tibble: 4 x 6
##   term              df   sumsq  meansq statistic   p.value
##   &lt;chr&gt;          &lt;int&gt;   &lt;dbl&gt;   &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;
## 1 gType              1  0.0692  0.0692      1.12  2.98e- 1
## 2 devStage           4 71.0    17.8       289.    6.69e-23
## 3 gType:devStage     4  0.689   0.172       2.80  4.44e- 2
## 4 Residuals         29  1.78    0.0615     NA    NA
```

Tests of overall effects of a factor controlling for the other one
---

class: middle
## Overall effects: compound tests (cont.)

### Without interaction

&lt;Big&gt;
&gt; `\(H_0: \tau_{KO}=0\)` (1 df) 
`\(H_0: \tau_{P2}=\tau_{P6}=\tau_{P10}=\tau_{4W}=0\)` (on average!, 4 df)


```r
tidy(anova(lm(gExp~gType+devStage,hitDat)))
```

```
## # A tibble: 3 x 6
##   term         df   sumsq  meansq statistic   p.value
##   &lt;chr&gt;     &lt;int&gt;   &lt;dbl&gt;   &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;
## 1 gType         1  0.0692  0.0692     0.924  3.44e- 1
## 2 devStage      4 71.0    17.8      237.     8.40e-24
## 3 Residuals    33  2.47    0.0749    NA     NA
```
Tests of overall effects of a factor controlling for the other one

&lt;small&gt;
The `\(t\)`-test in `lm` and the `\(F\)`-test (1 df) in `anova` for gType are not equivalent due to unbalancedness
---

class: middle

# Nested models

### These examples are just special cases of nested models 
For example: does development have a significant effect on gene expression?

&lt;font color=blue&gt; Compare the models with and without `devStage`!!  

&lt;font color=black&gt;

**Model 1**: gExp~gType

**Model 2**: gExp~gType + devStage

Mathematically:

**Model 1**: `\(Y_{ijk}=\theta + \tau_{KO}x_{KO,ijk} + \varepsilon\)`

**Model 2**: `\(Y_{ijk}=\theta + \tau_{KO}x_{KO,ijk} + \tau_{P2}x_{P2,ijk}+\tau_{P6}x_{P6,ijk}+\tau_{P10}x_{P10,ijk}+\tau_{4W}x_{4W,ijk}+ \varepsilon\)`

&lt;big&gt; `$$H_0: \tau_{P2}=\tau_{P6}=\tau_{P10}=\tau_{4W}=0$$`
&lt;small&gt; The `\(x_{DD,ijk}\)` are dummy variables (see [math handout](../SupportingMaterial/math_handout.html))
---
# More general!
![](L7_LinearModels_files/nested.png)
---
class: middle
### Nested models in R


```r
addReduced&lt;- lm(gExp ~ gType, data = hitDat)
addFull&lt;- lm(gExp ~ gType+devStage, data = hitDat)
anova(addReduced,addFull)
```

```
## Analysis of Variance Table
## 
## Model 1: gExp ~ gType
## Model 2: gExp ~ gType + devStage
##   Res.Df    RSS Df Sum of Sq      F    Pr(&gt;F)    
## 1     37 73.498                                  
## 2     33  2.473  4    71.024 236.92 &lt; 2.2e-16 ***
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
```

```r
tidy(anova(addFull))
```

```
## # A tibble: 3 x 6
##   term         df   sumsq  meansq statistic   p.value
##   &lt;chr&gt;     &lt;int&gt;   &lt;dbl&gt;   &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;
## 1 gType         1  0.0692  0.0692     0.924  3.44e- 1
## 2 devStage      4 71.0    17.8      237.     8.40e-24
## 3 Residuals    33  2.47    0.0749    NA     NA
```


```
## 
## Call:
## lm(formula = gExp ~ gType * devStage, data = twoDat)
## 
## Residuals:
##     Min      1Q  Median      3Q     Max 
## -0.3833 -0.1645 -0.1090  0.1298  0.5757 
## 
## Coefficients:
##                       Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)             9.9080     0.1575  62.911 2.03e-15 ***
## gTypeNrlKO             -0.9857     0.2406  -4.097  0.00177 ** 
## devStage4W              1.6345     0.2227   7.339 1.47e-05 ***
## gTypeNrlKO:devStage4W  -2.0381     0.3278  -6.217 6.56e-05 ***
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## Residual standard error: 0.315 on 11 degrees of freedom
## Multiple R-squared:  0.9507,	Adjusted R-squared:  0.9373 
## F-statistic: 70.76 on 3 and 11 DF,  p-value: 1.78e-07
```

---
class: middle
## Another special case: goodness of fit!

&lt;Big&gt; Compare the full *vs* the intercept-only models (compound test)!
`$$H_0: \tau_{KO}=\tau_{P2}=\tau_{P6}=\tau_{P10}=\tau_{4W}=0, \ (5 \text{ df})$$`


```
## Analysis of Variance Table
## 
## Model 1: gExp ~ 1
## Model 2: gExp ~ gType + devStage
##   Res.Df    RSS Df Sum of Sq      F    Pr(&gt;F)    
## 1     38 73.567                                  
## 2     33  2.473  5    71.094 189.72 &lt; 2.2e-16 ***
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
```


```r
summary(addFull)$fstatistic # also given in the summary of lm
```

```
##    value    numdf    dendf 
## 189.7238   5.0000  33.0000
```

&lt;small&gt; All is valid in the model with interaction, try it!!

---
class: middle
## Summary
&lt;big&gt;
- ** *t*-tests** can be used to test the equality of **2** population means

- **ANOVA** can be used to test the equality of **more than 2** population means simultaneously (main effects)

- **Linear regression** provides a general framework for modelling the relationship between a response and different type of explanatory variables

  - *t*-tests are used to test the significance of *simple effects* (*individual* coefficients)

  - *F*-tests are used to test the significance of *main effects* (*simultaneously* multiple coefficients) 

- *F*-tests are used to compare nested models
  - e.g., *overall* effects or *goodness of fit* 
---
class: middle

&lt;center&gt;

&lt;img src="L7_LinearModels_files/LM_vbles.png" width="1845" height="500px" /&gt;

&lt;/center&gt;

### Next class: linear models provides a general flexible framework to study the relation of a response with many variables!





    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"highlightSpans": true,
"countIncrementalSlides": false,
"ratio": "16:9"
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
