{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# Statistical Methods for High Dimensional Biology (STAT/BIOF/GSAT 540)\n",
    "\n",
    "\n",
    "# <font color=red> Lecture 6: ANOVA and Linear Models"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "subslide"
    }
   },
   "source": [
    "# Overview\n",
    "In this lecture, we will go beyond 2-groups comparisons and use ANOVA to compare these groups.\n",
    "\n",
    "### Table of Contents:\n",
    "\n",
    "1. [**Data description**](#Data-description)\n",
    "    1. [Loading the data](#Loading-the-data)\n",
    "    2. [Subsetting the data](#Subsetting-the-data)\n",
    "\n",
    "### R Dependencies:\n",
    "\n",
    "- knitr\n",
    "- lattice\n",
    "- latticeExtra\n",
    "- RColorBrewer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true,
    "slideshow": {
     "slide_type": "skip"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: 'dplyr'\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "Registered S3 methods overwritten by 'ggplot2':\n",
      "  method         from \n",
      "  [.quosures     rlang\n",
      "  c.quosures     rlang\n",
      "  print.quosures rlang\n",
      "\n",
      "Attaching package: 'gridExtra'\n",
      "\n",
      "The following object is masked from 'package:dplyr':\n",
      "\n",
      "    combine\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(knitr)\n",
    "knitr::opts_knit$set(root.dir=\".\")\n",
    "\n",
    "library(dplyr)\n",
    "\n",
    "library(ggplot2)\n",
    "library(ggthemes)\n",
    "\n",
    "library(grid)\n",
    "library(gridExtra)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# Data description\n",
    "\n",
    "Dataset was obtained from: [GSE4051](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE4051)\n",
    "\n",
    "> ## Targeting of GFP to new-born rods by Nrl promoter and temporal expression profiling of flow-sorted photoreceptors\n",
    "Akimoto M, Cheng H, Zhu D, Brzezinski JA, Khanna R, Filippova E, Oh EC, Jing Y, Linares JL, Brooks M, Zareparsi S, Mears AJ, Hero A, Glaser T, Swaroop A.\n",
    "<br><br>\n",
    "The Maf-family transcription factor Nrl is a key regulator of photoreceptor differentiation in mammals. Ablation of the Nrl gene in mice leads to functional cones at the expense of rods. We show that a 2.5-kb Nrl promoter segment directs the expression of enhanced GFP specifically to rod photoreceptors and the pineal gland of transgenic mice. GFP is detected shortly after terminal cell division, corresponding to the timing of rod genesis revealed by birthdating studies. In Nrl-/- retinas, the GFP+ photoreceptors express S-opsin, consistent with the transformation of rod precursors into cones. We report the gene profiles of freshly isolated flow-sorted GFP+ photoreceptors from wild-type and Nrl-/- retinas at five distinct developmental stages. Our results provide a framework for establishing gene regulatory networks that lead to mature functional photoreceptors from postmitotic precursors. Differentially expressed rod and cone genes are excellent candidates for retinopathies.\n",
    "\n",
    "### Overall Design\n",
    "> Postmitotic rod precursors and mature rod photoreceptors are tagged by GFP under the control of an Nrl promoter in the wild type background (Wt-Gfp mice). When cross-bred into the Nrl-knockout background (Nrl-ko-Gfp mice), the transformed “S-cones” are tagged by GFP. GFP positive photoreceptors from the Wt-Gfp or Nrl-ko-Gfp retina were enriched (purified) by FACS at five distinct developmental stages (E16, P2, P6, P10, and 4 weeks). Total RNA was extracted by Trizol regent and around 50 ng of total RNA was used for linear amplification and biotin labeling following Nugene kit protocol. Fragmented cDNA was hybridized on Affymetrix mouse genomic expression array 430 2.0 and then scanned with the standard protocol. Four replicates were performed for each time point."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "subslide"
    }
   },
   "source": [
    "# Loading the data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true,
    "slideshow": {
     "slide_type": "-"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "'data.frame':\t39 obs. of  4 variables:\n",
      " $ sidChar : chr  \"Sample_20\" \"Sample_21\" \"Sample_22\" \"Sample_23\" ...\n",
      " $ sidNum  : num  20 21 22 23 16 17 6 24 25 26 ...\n",
      " $ devStage: Factor w/ 5 levels \"E16\",\"P2\",\"P6\",..: 1 1 1 1 1 1 1 2 2 2 ...\n",
      " $ gType   : Factor w/ 2 levels \"wt\",\"NrlKO\": 1 1 1 1 2 2 2 1 1 1 ...\n",
      "'data.frame':\t29949 obs. of  39 variables:\n",
      " $ Sample_20: num  7.24 9.48 10.01 8.36 8.59 ...\n",
      " $ Sample_21: num  7.41 10.02 10.04 8.37 8.62 ...\n",
      " $ Sample_22: num  7.17 9.85 9.91 8.4 8.52 ...\n",
      " $ Sample_23: num  7.07 10.13 9.91 8.49 8.64 ...\n",
      " $ Sample_16: num  7.38 7.64 8.42 8.36 8.51 ...\n",
      " $ Sample_17: num  7.34 10.03 10.24 8.37 8.89 ...\n",
      " $ Sample_6 : num  7.24 9.71 10.17 8.84 8.54 ...\n",
      " $ Sample_24: num  7.11 9.75 9.39 8.37 8.36 ...\n",
      " $ Sample_25: num  7.19 9.16 10.11 8.2 8.5 ...\n",
      " $ Sample_26: num  7.18 9.49 9.41 8.73 8.39 ...\n",
      "  [list output truncated]\n"
     ]
    }
   ],
   "source": [
    "prDes <- readRDS(\"data/GSE4051_design.rds\")\n",
    "str(prDes)\n",
    "\n",
    "prDat<-read.table(\"data/GSE4051_data.tsv\",\n",
    "                  sep = \"\\t\", header = T, row.names = 1)\n",
    "str(prDat, list.len = 10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "subslide"
    }
   },
   "source": [
    "Here we examine the design of the study with a frequency table. We have four samples for each combination of development stage and genotype (except for \"E16 and NrlKO\", which has 3 samples)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "slideshow": {
     "slide_type": "-"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "         gType\n",
       "devStage  wt NrlKO\n",
       "  E16      4     3\n",
       "  P2       4     4\n",
       "  P6       4     4\n",
       "  P10      4     4\n",
       "  4_weeks  4     4"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "with(prDes, table(devStage, gType))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'WT samples = 20'"
      ],
      "text/latex": [
       "'WT samples = 20'"
      ],
      "text/markdown": [
       "'WT samples = 20'"
      ],
      "text/plain": [
       "[1] \"WT samples = 20\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "'NrlKO samples = 19'"
      ],
      "text/latex": [
       "'NrlKO samples = 19'"
      ],
      "text/markdown": [
       "'NrlKO samples = 19'"
      ],
      "text/plain": [
       "[1] \"NrlKO samples = 19\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Caculate total sample size of WT\n",
    "wtSample <- with(prDes, sum(gType == \"wt\"))\n",
    "paste(\"WT samples =\", wtSample)\n",
    "\n",
    "# Caculate total sample size of NrlKO\n",
    "nrlkoSample <- with(prDes, sum(gType == \"NrlKO\"))\n",
    "paste(\"NrlKO samples =\", nrlkoSample)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "subslide"
    }
   },
   "source": [
    "A cursory glance at the experession data reveals:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "slideshow": {
     "slide_type": "-"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Sample_20</th><th scope=col>Sample_21</th><th scope=col>Sample_22</th><th scope=col>Sample_23</th><th scope=col>Sample_16</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1415670_at</th><td> 7.236</td><td> 7.414</td><td>7.169</td><td> 7.070</td><td>7.383</td></tr>\n",
       "\t<tr><th scope=row>1415671_at</th><td> 9.478</td><td>10.020</td><td>9.854</td><td>10.130</td><td>7.637</td></tr>\n",
       "\t<tr><th scope=row>1415672_at</th><td>10.010</td><td>10.040</td><td>9.913</td><td> 9.907</td><td>8.423</td></tr>\n",
       "\t<tr><th scope=row>1415673_at</th><td> 8.362</td><td> 8.374</td><td>8.404</td><td> 8.487</td><td>8.363</td></tr>\n",
       "\t<tr><th scope=row>1415674_a_at</th><td> 8.585</td><td> 8.615</td><td>8.520</td><td> 8.641</td><td>8.509</td></tr>\n",
       "\t<tr><th scope=row>1415675_at</th><td> 9.591</td><td> 9.719</td><td>9.709</td><td> 9.700</td><td>9.656</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & Sample\\_20 & Sample\\_21 & Sample\\_22 & Sample\\_23 & Sample\\_16\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1415670\\_at &  7.236 &  7.414 & 7.169 &  7.070 & 7.383\\\\\n",
       "\t1415671\\_at &  9.478 & 10.020 & 9.854 & 10.130 & 7.637\\\\\n",
       "\t1415672\\_at & 10.010 & 10.040 & 9.913 &  9.907 & 8.423\\\\\n",
       "\t1415673\\_at &  8.362 &  8.374 & 8.404 &  8.487 & 8.363\\\\\n",
       "\t1415674\\_a\\_at &  8.585 &  8.615 & 8.520 &  8.641 & 8.509\\\\\n",
       "\t1415675\\_at &  9.591 &  9.719 & 9.709 &  9.700 & 9.656\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 5\n",
       "\n",
       "| <!--/--> | Sample_20 &lt;dbl&gt; | Sample_21 &lt;dbl&gt; | Sample_22 &lt;dbl&gt; | Sample_23 &lt;dbl&gt; | Sample_16 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| 1415670_at |  7.236 |  7.414 | 7.169 |  7.070 | 7.383 |\n",
       "| 1415671_at |  9.478 | 10.020 | 9.854 | 10.130 | 7.637 |\n",
       "| 1415672_at | 10.010 | 10.040 | 9.913 |  9.907 | 8.423 |\n",
       "| 1415673_at |  8.362 |  8.374 | 8.404 |  8.487 | 8.363 |\n",
       "| 1415674_a_at |  8.585 |  8.615 | 8.520 |  8.641 | 8.509 |\n",
       "| 1415675_at |  9.591 |  9.719 | 9.709 |  9.700 | 9.656 |\n",
       "\n"
      ],
      "text/plain": [
       "             Sample_20 Sample_21 Sample_22 Sample_23 Sample_16\n",
       "1415670_at    7.236     7.414    7.169      7.070    7.383    \n",
       "1415671_at    9.478    10.020    9.854     10.130    7.637    \n",
       "1415672_at   10.010    10.040    9.913      9.907    8.423    \n",
       "1415673_at    8.362     8.374    8.404      8.487    8.363    \n",
       "1415674_a_at  8.585     8.615    8.520      8.641    8.509    \n",
       "1415675_at    9.591     9.719    9.709      9.700    9.656    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(subset(prDat, select = 1:5))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "subslide"
    }
   },
   "source": [
    "# Subsetting the data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "subslide"
    }
   },
   "source": [
    "We extract the rows for each of these genes, transpose and then vectorize the data. We then merge this data together with the design metadata. Ensure that the samples match between the data and design before merging."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in data.frame(prDes, miniDat):\n",
      "\"row names were found from a short variable and have been discarded\""
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "'data.frame':\t78 obs. of  6 variables:\n",
      " $ sidChar : chr  \"Sample_20\" \"Sample_21\" \"Sample_22\" \"Sample_23\" ...\n",
      " $ sidNum  : num  20 21 22 23 16 17 6 24 25 26 ...\n",
      " $ devStage: Factor w/ 5 levels \"E16\",\"P2\",\"P6\",..: 1 1 1 1 1 1 1 2 2 2 ...\n",
      " $ gType   : Factor w/ 2 levels \"wt\",\"NrlKO\": 1 1 1 1 2 2 2 1 1 1 ...\n",
      " $ gene    : Factor w/ 2 levels \"Irs4\",\"Nrl\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      " $ gExp    : num  7.87 7.78 7.58 7.4 8.08 ...\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>sidChar</th><th scope=col>sidNum</th><th scope=col>devStage</th><th scope=col>gType</th><th scope=col>gene</th><th scope=col>gExp</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Sample_20</td><td>20</td><td>E16</td><td>wt   </td><td>Irs4</td><td>7.867</td></tr>\n",
       "\t<tr><td>Sample_21</td><td>21</td><td>E16</td><td>wt   </td><td>Irs4</td><td>7.783</td></tr>\n",
       "\t<tr><td>Sample_22</td><td>22</td><td>E16</td><td>wt   </td><td>Irs4</td><td>7.585</td></tr>\n",
       "\t<tr><td>Sample_23</td><td>23</td><td>E16</td><td>wt   </td><td>Irs4</td><td>7.400</td></tr>\n",
       "\t<tr><td>Sample_16</td><td>16</td><td>E16</td><td>NrlKO</td><td>Irs4</td><td>8.076</td></tr>\n",
       "\t<tr><td>Sample_17</td><td>17</td><td>E16</td><td>NrlKO</td><td>Irs4</td><td>7.515</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       " sidChar & sidNum & devStage & gType & gene & gExp\\\\\n",
       " <chr> & <dbl> & <fct> & <fct> & <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t Sample\\_20 & 20 & E16 & wt    & Irs4 & 7.867\\\\\n",
       "\t Sample\\_21 & 21 & E16 & wt    & Irs4 & 7.783\\\\\n",
       "\t Sample\\_22 & 22 & E16 & wt    & Irs4 & 7.585\\\\\n",
       "\t Sample\\_23 & 23 & E16 & wt    & Irs4 & 7.400\\\\\n",
       "\t Sample\\_16 & 16 & E16 & NrlKO & Irs4 & 8.076\\\\\n",
       "\t Sample\\_17 & 17 & E16 & NrlKO & Irs4 & 7.515\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| sidChar &lt;chr&gt; | sidNum &lt;dbl&gt; | devStage &lt;fct&gt; | gType &lt;fct&gt; | gene &lt;fct&gt; | gExp &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| Sample_20 | 20 | E16 | wt    | Irs4 | 7.867 |\n",
       "| Sample_21 | 21 | E16 | wt    | Irs4 | 7.783 |\n",
       "| Sample_22 | 22 | E16 | wt    | Irs4 | 7.585 |\n",
       "| Sample_23 | 23 | E16 | wt    | Irs4 | 7.400 |\n",
       "| Sample_16 | 16 | E16 | NrlKO | Irs4 | 8.076 |\n",
       "| Sample_17 | 17 | E16 | NrlKO | Irs4 | 7.515 |\n",
       "\n"
      ],
      "text/plain": [
       "  sidChar   sidNum devStage gType gene gExp \n",
       "1 Sample_20 20     E16      wt    Irs4 7.867\n",
       "2 Sample_21 21     E16      wt    Irs4 7.783\n",
       "3 Sample_22 22     E16      wt    Irs4 7.585\n",
       "4 Sample_23 23     E16      wt    Irs4 7.400\n",
       "5 Sample_16 16     E16      NrlKO Irs4 8.076\n",
       "6 Sample_17 17     E16      NrlKO Irs4 7.515"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "miniDat <- as.vector(t(prDat[c(\"1422248_at\", \"1450946_at\"), ]))\n",
    "miniDat <- data.frame(gene = rep(c(\"Irs4\", \"Nrl\"), each = nrow(prDes)),\n",
    "                      gExp = miniDat)\n",
    "miniDat <- data.frame(prDes, miniDat) # ignore the warning about row names\n",
    "str(miniDat)\n",
    "head(miniDat)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Code from Lecture 6"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Slide 5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "\tTwo Sample t-test\n",
       "\n",
       "data:  gExp by gType\n",
       "t = 0.52865, df = 37, p-value = 0.6002\n",
       "alternative hypothesis: true difference in means is not equal to 0\n",
       "95 percent confidence interval:\n",
       " -0.07383844  0.12597002\n",
       "sample estimates:\n",
       "   mean in group wt mean in group NrlKO \n",
       "           7.765750            7.739684 \n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "t.test(gExp ~ gType, miniDat, subset = gene == \"Irs4\", var.equal = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "            Df Sum Sq Mean Sq F value Pr(>F)\n",
       "gType        1 0.0066 0.00662   0.279    0.6\n",
       "Residuals   37 0.8764 0.02369               "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "summary(aov(gExp ~ gType, miniDat, subset = gene == \"Irs4\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "lm(formula = gExp ~ gType, data = miniDat, subset = gene == \"Irs4\")\n",
       "\n",
       "Residuals:\n",
       "     Min       1Q   Median       3Q      Max \n",
       "-0.36575 -0.09072  0.01725  0.09428  0.36525 \n",
       "\n",
       "Coefficients:\n",
       "            Estimate Std. Error t value Pr(>|t|)    \n",
       "(Intercept)  7.76575    0.03441 225.650   <2e-16 ***\n",
       "gTypeNrlKO  -0.02607    0.04931  -0.529      0.6    \n",
       "---\n",
       "Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1\n",
       "\n",
       "Residual standard error: 0.1539 on 37 degrees of freedom\n",
       "Multiple R-squared:  0.007497,\tAdjusted R-squared:  -0.01933 \n",
       "F-statistic: 0.2795 on 1 and 37 DF,  p-value: 0.6002\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "summary(lm(gExp ~ gType, miniDat, subset = gene == \"Irs4\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Slide 23"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "theHit <- which(rownames(prDat) == \"1440645_at\") # 17843\n",
    "## and this as our boring gene\n",
    "theBore <- which(rownames(prDat) == \"1443184_at\") # 18898\n",
    "\n",
    "keepers <- data.frame(row = c(theBore, theHit),\n",
    "                       probesetID = I(rownames(prDat)[c(theBore, theHit)]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in data.frame(prDes, miniDat):\n",
      "\"row names were found from a short variable and have been discarded\""
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "'data.frame':\t78 obs. of  6 variables:\n",
      " $ sidChar : chr  \"Sample_20\" \"Sample_21\" \"Sample_22\" \"Sample_23\" ...\n",
      " $ sidNum  : num  20 21 22 23 16 17 6 24 25 26 ...\n",
      " $ devStage: Factor w/ 5 levels \"E16\",\"P2\",\"P6\",..: 1 1 1 1 1 1 1 2 2 2 ...\n",
      " $ gType   : Factor w/ 2 levels \"wt\",\"NrlKO\": 1 1 1 1 2 2 2 1 1 1 ...\n",
      " $ gene    : Factor w/ 2 levels \"theBore\",\"theHit\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      " $ gExp    : num  7.59 7.96 7.43 7.58 7.98 ...\n"
     ]
    }
   ],
   "source": [
    "miniDat <- as.vector(t(prDat[keepers$probesetID, ]))\n",
    "miniDat <- data.frame(gene = rep(c(\"theBore\", \"theHit\"), each = nrow(prDes)),\n",
    "                      gExp = miniDat)\n",
    "\n",
    "miniDat <- data.frame(prDes, miniDat)\n",
    "str(miniDat)\n",
    "\n",
    "boreDat <- filter(miniDat, gene == \"theBore\")\n",
    "hitDat <- filter(miniDat, gene == \"theHit\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAMAAADKOT/pAAAAV1BMVEUAAAADAwMHBwcODg4P\nDw8dHR0fHx8zMzM6Ojo/Pz9NTU1oaGh1dXV8fHx/f3+MjIyampqnp6eysrK9vb3Hx8fQ0NDZ\n2dnh4eHp6enr6+vw8PD/AAD////hsE7hAAAACXBIWXMAABJ0AAASdAHeZh94AAAgAElEQVR4\nnO2dbUPaWLgAufd2reBL293tWlv+/++8JiASCC9PcpIz+Mx8UESN0yTDSQ4pLNYiMppFbQGR\nz4AhiRTAkEQKYEgiBTAkkQIYkkgBDEmkAIYkUgBDEimAIYkUYHBIv7+3v37i9xcbHv4euniR\nm2JwSJuELoS0WDwNXb7ILTFZSO2nl6fFj6F/QOSGmDak9a/FaugfELkhhobUHre1wfyzWjz9\n197339PbodzP7bd3P9d+evm+Wqy+v2zu+f2w+Nb9cZEbZ3RI39tbTUn/bs6Kfmy+vfmx34uH\n5tPP7RnTz/Zb39of2v9xkRtn9KHd6uf697dmTuGlHZjezop+rvfPkZqvXhaLv3+vf/9YLH41\n33r6vT74cZEbZ3RITQe/my9+LJo+3m5/W+/N2rXz3z+248735vNm+Or+uMiNU2ayofn0sItn\nvRfSt+a86KEZidbN1MPD7lc6Py5y45QLadENqb3/9z+rZvzpTD3sfsWQ5PNQMqSj767bU6FT\nIQ39yyI8yoW0Wrwcfvf9Vt+hXefHRW6cciF9X7RX37VD0P6ItGomFtpvvU82bG/v/bjIjTMi\npF/rTkgv7XNCL6v96e/1f6vmzl9v39pMf798zIzv/7jIjTM4pIfF4QnPz84Tsp2rVrtPyG4W\nsP/jIjfO4JBeHpqDts7Mwa8fq8XDP9s7trPf/25/ev8Soe0S9n5c5MZx7kykAIYkUgBDEimA\nIYkUwJBECmBIIgUwJJECGJJIAQxJpACGJFIAQxIpwNCQvorI1/EhDfy9i7xOteAxIKWYVrmk\nDCkGUopplUvKkGIgpZhWuaQMKQZSimmVS8qQYiClmFa5pAwpBlKKaZVLypBiIKWYVrmkDCkG\nUopplUvKkGIgpZhWuaQMKQZSimmVSyoW0ubdK1dv9C2hMLk2xCiQVrmkQiFt+vn4cLCEwuTa\nEKNAWuWSioS0WhsSUopplUtqwKGdIeFAWuWSGhXS5vrxV5H0OCLFQEoxrXJJGVIMpBTTKpeU\nIcVASjGtckkZUgykFNMql5QhxUBKMa1ySXllQwykFNMql5TX2sVASjGtckkZUgykFNMql5Qh\nxUBKMa1ySRlSDKQU0yqXlCHFQEoxrXJJGVIMpBTTKpeUIcVASjGtckkZUgykFNMql5QhxUBK\nMa1ySRlSDKQU0yqXlCHFQEoxrXJJGVIMpBTTKpeUIcVASjGtckkZUgykFNMql5QhxUBKMa1y\nSRlSDKQU0yqXlCHFQEoxrXJJGVIMpBTTKpeUIcVASjGtckkZUgykFNMql5QhxUBKMa1ySRlS\nDKQU0yqXlCHFQEoxrXJJFQip9vsAiNTHESkGUopplUvKkGIgpZhWuaQMKQZSimmVS8qQYiCl\nmFa5pAwpBlKKaZVLypBiIKWYVrmkDCkGUopplUvKkGIgpZhWuaQMKQZSimmVS8qQYiClmFa5\npAwpBlKKaZVLypBiIKWYVrmkDCkGUopplUvKkGIgpZhWuaQMKQZSimmVS8qQYiClmFa5pAwp\nBlKKaZVLypBiIKWYVrmkDCkGUopplUvKkGIgpZhWuaQMKQZSimmVS8qQYiClmFa5pAwpBlKK\naZVLypBiIKWYVrmkhoS0eqNvCYXJtSFGgbTKJTUgpNXuw8ESCpNrQ4wCaZVLypBiIKWYVrmk\nDCkGUopplUtqVEhfW2q/fLlIfQaGtHJEIoG0yiU1cNbOQzsUSKtcUkOfRzIkEkirXFJONsRA\nSjGtckkZUgykFNMql5RXNsRASjGtckl5rV0MpBTTKpeUIcVASjGtckkZUgykFNMql5QhxUBK\nMa1ySRlSDKQU0yqXlCHFQEoxrXJJGVIMpBTTKpeUIcVASjGtckkZUgykFNMql5QhxUBKMa1y\nSRlSDKQU0yqXlCHFQEoxrXJJGVIMpBTTKpeUIcVASjGtckkZUgykFNMql5QhxUBKMa1ySRlS\nDKQU0yqXlCHFQEoxrXJJGVIMpBTTKpeUIcVASjGtckkZUgykFNMql5QhxUBKMa1ySRlSDKQU\n0yqXVIGQar8PgEh9HJFiIKWYVrmkDCkGUopplUvKkGIgpZhWuaQMKQZSimmVS8qQYiClmFa5\npAwpBlKKaZVLypBiIKWYVrmkDCkGUopplUvKkGIgpZhWuaQMKQZSimmVS8qQYiClmFa5pAwp\nBlKKaZVLypBiIKWYVrmkDCkGUopplUvKkGIgpZhWuaQMKQZSimmVS8qQYiClmFa5pAwpBlKK\naZVLypBiIKWYVrmkDCkGUopplUvKkGIgpZhWuaQMKQZSimmVS8qQYiClmFa5pAwpBlKKaZVL\nypBiIKWYVrmkhoS0eqNvCYXJtSFGgbTKJTUgpNXuw8ESCpNrQ4wCadUvtVwuZ/boYEjr6tug\nBbnLMq16pZbPz881t2LWkPbbqb0NWpC7LNOqT6rZhlW3IjOkry0TvsT//ePj4/3e7Y+v5Cb5\nvBsRPdnQefyq/mDWgnzsZ1o5Il1gvkM7Q7oSpJXnSBeoFFL1bdCC3GWZVs7aXWDGyYZuO87a\nnQJplUuKHRKinQ7V947eNVLdqo9cUujJBiC1pfqPb2tb9cKWOnpAGvmY7bV2F+ms4cpSJ2Zc\nKKuqA1rq6AFp7Bm4IV2iu4YN6WrIUkfrcfScsCFd4GANG9LVkKUMaXYYIe0OLz1HGkeFkH7/\neFgsnv4OLKEwjA2BCGkvH2ftRjH/OdKv1aJl9evqJRQGsiEA50gXHy4hq6oLW2q2WbunxdNb\nQr+eFt+vXkJhKBui/qydIRVj/ueRFpuvfy8unTt9+pA6GNLV5JI6GdK3xe/N3U9XL6EwuTbE\nWS4dwHNXFe3SlApXNnx/emkO7Z6ynyN1qT1r189Jq5r7cXuJF+FC431qHNrtc9USCmNIV3PK\nqup+/Er5ry/7GNKsnH4gv6mQ6u7HhjR0CYWpt8+eeSA3pKsxpKFLKEy1ffbc5jekqwGcI22O\nLPaPL+YPqZlqaPj97eolFKbay3F9mpCqnyNVnrXb/PM7K6HGOdKP5tPfkz6PdHY1H/6bZ9sr\ntiHtyX3cvK2Q5tiPO39ilgf/a9lsxy+dh8X5Q3p5Wqz+/Xe1eHi5eglhzqdx8G+e8Til9dq/\nvu3jZvW9o5ea8zL7G2WeB/9rYYS0Xr9VtHj4GVhClAtp1AupeWDd+3O7m2/3Vt87eqk7L3Pi\nhZ5eaz8bSwnpZ7KQ/szMYNHj/bPg7jH3WriKgf8WxDnSy8Pkh3axkE4cCE61tT7kll/unt+P\n9Z4fy5VczrwkwX/EmRHpvtwjX7l/Xo3Jhva/Ik062XD+HGmSDR+Xe/t8d/d89+UopEoHLn0P\nP8hzpIIhjXHqUuGi1c01q9NOf4dm7eZm+yREszfc/c/d9mD7sX+nmVOrCfsOE9LpWbtqIVV5\n+sInZC+wCenu/bT1vnN/nd3k7m13/dK573hVAa66fu15qJlFCxTS3vHc5UO7iV7e//4e8aYF\n7fsn/PVX+yYK+0r13lfh/940/rrwl+8R7/lwtAnn0aqyac6EtEmo1n/sWz4/Mi7SWnamUF/3\n7650bHfXDJCdv9w3L1P9GreeYXImrQHnSKOHSmpIb+visfqusKFZx7tN87Ehql1+07M7GtLB\nH4pevT9+YxrSdbxvmtfju+Z3uThrVzmkzZqpGNJp+kMqIGZIMRAzIEcJH1/fW7WjzR/vWVX1\n/8NsupA450hdECEdgZq1e98r+1ZV9cnEfCGtl/fAjvghVd9Vz4ZUnRMhfTmcvgnDDWn9Wn+f\nOAaydxysGsAUyAenQwJs0N7tt7y7+98vE83aXfd6Dd0lFOa+/j5xDCOkw1wAk/J7nDpHAkTe\nf7zZPMN992lDWj4C9okjECEd5cIK6cSsHcKtb/t9af7XzMG1ImG4lwgZ0knoIW0wpKFLKIsh\nneR0SIjDpy03E9KEh3br/aO71fezL7bqOdL8nDxHQpzQb7mZc6T2oquxYte9QOS5kpy1q8DJ\nWTsQtzNrV0LsZEg/39/W5d/1j7Nv7fLJ/xvFAUgpplUuqZMhPby/G8XDheeSDKk+SKtcUpfe\nH6n9bEg7kFJMq1xSl9+x72n9XzMqXbGEwuTaEKNAWuWSuuY9ZBeLf65aQmFybYhRIK1ySZ15\nV/O/HxaLhx9vZ0qbFy++vITC5NoQo0Ba5ZLiPiGbbEOMAmmVS8qQYiClmFa5pAwpBlKKaZVL\nypBiIKWYVrmkDCkGUopplUtqQEirlp4lFCbXhhgF0iqX1NARyZBIIK1ySQ0M6aMjQwKAtMol\nZUgxkFJMq1xSw0LadvS1Zd4XKxchMiqkgxQLc/bBo9b/EUM+zDKtckndaEjV/tcycu9gWuWS\nGhTSfkdVQqr3OhrIvYNplUvKkGIg9w6mVS4pQ4qB3DuYVrmkbjMkz5G6IK1ySd3qtXbO2u2D\ntMoldash1QIpxbTKJWVIMZBSTKtcUoYUAynFtMolZUgxkFJMq1xShhQDKcW0yiVlSDGQUkyr\nXFKGFAMpxbTKJWVIMZBSTKtcUoYUAynFtMolZUgxkFJMq1xShhQDKcW0yiVlSDGQUkyrXFKG\nFAMpxbTKJWVIMZBSTKtcUoYUAynFtMolZUgxkFJMq1xShhQDKcW0yiVlSDGQUkyrXFKGFAMp\nxbTKJWVIMZBSTKtcUoYUAynFtMolZUgxkFJMq1xSBUKq/T4AIvVxRIqBlGJa5ZIypBhIKaZV\nLilDioGUYlrlkjKkGEgpplUuKUOKgZRiWuWSMqQYSCmmVS4pQ4qBlGJa5ZIypBhIKaZVLilD\nioGUYlrlkjKkGEgpplUuKUOKgZRiWuWSMqQYSCmmVS4pQ4qBlGJa5ZIypBhIKaZVLilDioGU\nYlrlkjKkGEgpplUuKUOKgZRiWuWSMqQYSCmmVS4pQ4qBlGJa5ZIypBhIKaZVLilDioGUYlrl\nkjKkGEgpplUuKUOKgZRiWuWSMqQYSCmmVS4pQ4qBlGJa5ZIaEtLqjb4lFCbXhhgF0iqX1ICQ\nVrsPB0soTK4NMQqkVS4pQ4qBlGJa5ZIaGlLfEgqTa0OMAmmVS2pQSO/nSF9bar8PgEh9hoS0\nWntoxwJplUvKc6QYSCmmVS4pQ4qBlGJa5ZIypBhIKaZVLilDioGUYlrlkvLKhhhIKaZVLimv\ntYuBlGJa5ZIypBhIKaZVLilDioGUYlrlkjKkGEgpplUuKUOKgZRiWuWSMqQYSCmmVS4pQ4qB\nlGJa5ZIypBhIKaZVLilDioGUYlrlkjKkGEgpplUuKUOKgZRiWuWSMqQYSCmmVS4pQ4qBlGJa\n5ZIypBhIKaZVLilDioGUYlrlkjKkGEgpplUuKUOKgZRiWuWSMqQYSCmmVS4pQ4qBlGJa5ZIy\npBhIKaZVLilDioGUYlrlkjKkGEgpplUuqQIh1X75cpH6OCLFQEoxrXJJGVIMpBTTKpeUIcVA\nSjGtckkZUgykFNMql5QhxUBKMa1ySRlSDKQU0yqXlCHFQEoxrXJJGVIMpBTTKpeUIcVASjGt\nckkZUgykFNMql5QhxUBKMa1ySRlSDKQU0yqXlCHFQEoxrXJJGVIMpBTTKpeUIcVASjGtckkZ\nUgykFNMql5QhxUBKMa1ySRlSDKQU0yqXlCHFQEoxrXJJGVIMpBTTKpeUIcVASjGtckkZUgyk\nFNMql5QhxUBKMa1ySRlSDKQU0yqXlCHFQEoxrXJJDQlp1dCzhMLk2hCjQFrlkhoU0oklFCbX\nhhgF0iqXlCHFQEoxrXJJDQip25EhAUBa5ZIaEtLuFOlrS+33ARCpz9ARyckGEkirXFJDp78N\niQTSKpeUIcVASjGtckl5aBcDKcW0yiU1MKS9mTtDqg/SKpfU0Csb+pZQmFwbYhRIq1xSXmsX\nAynFtMolZUgxkFJMq1xShhQDKcW0yiVlSDGQUkyrXFKGFAMpxbTKJWVIMZBSTKtcUoYUAynF\ntMolZUgxkFJMq1xSBUISkfEhTcZkQ90YkFJMq6xShnQNSCmmVVYpQ7oGpBTTKqsULiSRW8SQ\nRApgSCIFMCSRAhiSSAEMSaQAiJBWWzZf7O6qqbTeWWxFGEItOKv9zVd1+135J6cwY4S0f/vj\n5Yrq7yIf+wZEqAVntZ9Qze13bbsZQlqtDekSOKu9kGpuv5Uj0tEX9XeOls7RCtAKJ7Re1wup\nuz46GgdHwod3lwAbEuPgv/OJt98SVxMopNXBILn/ufR4zgjpaLLh44GjKp2zaIDPhg8r4mqq\nFdLhqlhtEzoIaO+uon6MkI6+gBzfrc58VY9V91Z9LcL2O/qL+yFtO999fh88Cxoa0jmOxQgY\nUp9D5+Bh3TmuOzykWO2mJRKcIx3eXYXVidt1MaRrRM6FtD6+byyGdI7DWSAGhnS9yEc0OScb\nKE/Zf9xaMYwa9vNGSPXtv/WvbOjMXeWY/ha5dQxJpACGJJ+K1dH03TwYkkgBDEmkAIYkUgBD\nEimAIYkUwJBECmBIIgUwJJECGJJIAQxJpACGJFIAQxIpgCGJFMCQRApgSCIFMCSRAhiSSAEM\nSaQAhiRSAEMSKYAhiRTAkEQKYEgiBTAkkQIYkkgBDEmkAIYkUgBDEimAIYkUwJBECmBIIgUw\nJJECGJJIAQxJpACGJFIAQxIpgCGJFMCQRApgSCIFMCSRAhiSSAEMSaQAg0P6/b399RO//3F/\ne6v9sPkNkc/I4JA2qURCOvWzIrfPPCGtz/2syO1jSCIFGLpzLxraOP5ZLZ7+a+/772mxePq5\n/fbHD24/bH9D5DMyOqTv7a2mpH/bW4sfm29//ODakOTTM/rQbvVz/fvb4mm9fmkHppenxc91\nb0ge2sknZnRITTa/my9+LH6v29vf1tvhZ8vakOTTU2ayofn00C3HkCQT5UI6Lufj5wxJPjkl\nQzr67u6WIcknp1xIq8XL4Xd3twxJPjnlQvq+aK+le2lm8AxJkjEipF/rTkgv7VNIL6sz09+/\nRsqKUBkc0sNi78So/fTz/BOy298Q+YwM3rVfHharbkjrXz9Wi4d/9u7Y3Wo/bH5D5DPiGCFS\nAEMSKYAhiRTAkEQKYEgiBTAkkQIYkkgBDEmkAIYkUgBDEimAIYkUYGhIX0Xk6/iQBv7eRV6n\nWvAYkFJMq1xShhQDKcW0yiVlSDGQUkyrXFKGFAMpxbTKJWVIMZBSTKtcUoYUAynFtMolZUgx\nkFJMq1xShhQDKcW0yiVlSDGQUkyrXFKGFAMpxbTKJRULafMiWqs3+pZQmFwbYhRIq1xSoZA2\n/Xx8OFhCYXJtiFEgrXJJRUJarQ0JKcW0yiU14NDOkHAgrXJJjQppc/34q0h6HJFiIKWYVrmk\nDCkGUopplUvKkGIgpZhWuaQMKQZSimmVS8qQYiClmFa5pLyyIQZSimmVS8pr7WIgpZhWuaQM\nKQZSimmVS8qQYiClmFa5pAwpBlKKaZVLypBiIKWYVrmkDCkGUopplUvKkGIgpZhWuaQMKQZS\nimmVS8qQYiClmFa5pAwpBlKKaZVLypBiIKWYVrmkDCkGUopplUvKkGIgpZhWuaQMKQZSimmV\nS8qQYiClmFa5pAwpBlKKaZVLypBiIKWYVrmkDCkGUopplUvKkGIgpZhWuaQMKQZSimmVS8qQ\nYiClmFa3JrVcLkcs2JBiIKWYVtWkzhVxWmr5/Pw8oiRDioGUYlrVkjpbxEmp5rfGlFQgpNrv\nAyCyx/1jw/1Mv7bDESkGUoppVUnq/NBCHpEG/+0LuHdcDdLqtkLyHGlekFJMK8Q50uHEg7N2\nEJBSTCvCrN3RMOPzSBCQUkwrgNTxYd6B1LhRaB9DioGUYloBpM6GtFx++XI36rxoH0OKgZRi\nWgGkzoW0fL778mVZrCRDioGUYloRpE6fI711tPxy15RkSDVASjGtEFInZ+0MqS5IKaYVW6oJ\nadkc23mOVAWkFNMKLtWUdPc2JBVasCHFQEoxrehSbwNSsclvQ4qClGJa5ZIypBhIKaZVLilD\nioGUYlrlkjKkGEgpplUuKUOKgZRiWuWSMqQYSCmmVS4pQ4qBlGJa5ZIypBhIKaYVUeqPIUFA\nSjGtgFJ/HJEoIKWYVjypPx7aYUBKMa1wUm8dGRIFpBTTKpfUkJBWb/QtoTC5NsQokFY0qWZA\nQoW02n04WEJhaBuiBSnFtIJJtR0ZEgWkFNOKJbXpyJAoIKWYViipbUfQkL62DH7dcZGZ+DP5\nXxgY0soRiQTSCiT1Z3eLNCI1s3Ye2qFAWnGkPjpihdRgSCSQVhipvY5QITnZgANpRZHa78iQ\nKCClmFa5pLyyIQZSimkFkeoMSKyQTi2hMJAN0QUpxbRiSHU7MiQKSCmmFULqoCNDooCUYloR\npA47MiQKSCmmFUDqqCNDooCUYlrVlzruyJAoIKWYVtWlejoyJApIKaZVbam+jgyJAlKKaZVL\nypBiIKWYVpWlegckQ6KAlGJa1ZXq78iQKCClmFZVpU50ZEgUkFJMq5pSpzoyJApIKaZVRamT\nHRkSBaQU06qe1OmODIkCUoppVU3qTEeGRAEpxbSqJdXb0XK5bD8bEgSkFNOqklR/R8/Pz21J\nhgQBKcW0Akk1HW1KMiQISCmmVR2pkwOSIaFASjGtqkj1TzQYEg6kFNOqhtSpCTvPkWggpZhW\nFaROT3w7awcDKcW0ml/q3BNIW8ghTf5C/yLXMP1bTpzBESkGUoppNbfUFeMRe0QqI3KMe8fV\nIK1mlrqqI0OigJRiWs0rdV1HhkQBKcW0Aku9z+EVxJBiIKWYVrNKXTkgbaR2zyoVxJBiIKWY\nVnNKXdtRK/VxnUNBDCkGUoppNaPU1R0ZEgWkFNNqPqnrOzIkCkgpptVsUoGOPEeigJRiWs0l\nFenIWTsKSCmm1UxSoY58HokCUoppNY9UrCNDooCUYlrNIhXsyJAoIKWYVrmkDCkGUoppNYdU\ndEAyJApIKabVDFLhjgyJAlKKaTW9VLwjQ6KAlGJaTS41oCNDooCUYlpNLTWkI0OigJRiWk0s\nNagjQ6KAlGJaTSs1rCNDooCUYlpNKjWwI0OigJRiWuWSMqQYSCmm1ZRSQwckQ6KAlGJaTSg1\nuCNWSKs3+pZQmGx7xwiQVtNJDe8IFdJq9+FgCYVJtneMAWk1mdSIjgyJAlKKaTWV1JiODIkC\nUoppNZHUqI6gIX1tqfkeAJKNqm85cQYnG2IgpZhWk0iNG4+oI9LBEgqTZ+8YDdJqCqmxHRkS\nBaQU0yqXlCHFQEoxrSaQGj0gGRIFpBTTqrzU+I5QITnZgANpVVyqQEeskE4toTA59o4iIK1K\nS5XoyJAoIKWYVoWlinRkSBSQUkyrslJlOjIkCkgpplVRqUIdGRIFpBTTqqRUqY4MiQJSimlV\nUKpYR4ZEASnFtMolZUgxkFJMq3JS5QYkQ6KAlGJaFZMq2JEhUUBKMa1KSZXsyJAoIKWYVoWk\ninZkSBSQUkyrMlJlOzIkCkgpplURqcIdGRIFpBTTqoRU6Y4MiQJSimlVQKp4R4ZEASnFtBov\nVb4jQ6KAlGJa5ZIypBhIKabVaKkJBiRDooCUYlqNlZqiI0OigJRiWo2UmqQjQ6KAlGJajZOa\npiNDooCUYlqNkpqoI0OigJRiWo2RmqojdEi13wdAPh3Ut5w4gyNSDKQU02q41GTjEXtEKiNy\nzCfbO6YEaTVYasKODIkCUopplUvKkGIgpZhWQ6WmHJAMiQJSimk1UGrSjgyJAlKKaTVMatqO\nDIkCUoppNUhq4o4MiQJSimk1RGrqjgyJAlKKaTVAavKODIkCUoppFZeaviNDooCUYlqFpWbo\nyJAoIKWYVrmkDCkGUoppFZWaY0AyJApIKaZVUGqWjgyJAlKKaRWTmqcjQ6KAlGJahaRm6siQ\nKCClmFYRqbk6MiQKSCmmVUBqto4MiQJSiml1vdR8HRkSBaQU0+pqqRk7MiQKSCmm1bVSc3Zk\nSBSQUkyrXFKGFAMpxbS6UmrWAcmQKCClmFbXSc3bESqkVUvPEgpzw3vH3CCtrpKauSNUSC2G\nRAJpdY3U3B3hQvroyJAAIK2ukJq9I0OigJRiWl2Wmr8jWkjbjr621H75crlRbvCl8k8zKqSD\nFAtzow+zNUBaXZKqMB5RR6SDJRTmJveOOiCtLkhV6QgW0n5HhgQAaZVLypBiIKWYVuel6gxI\nhkQBKcW0OitVqSNDooCUYlqdk6rVESykE0sozM3tHfVAWp2RqtaRIVFASjGtTkvV68iQKCCl\nmFYnpSp2ZEgUkFJMq1NSNTsyJApIKabVCamqHRkSBaQU06pfqm5HhkQBKcW0yiVlSDGQUkyr\nXqnKA5IhUUBKMa36pGp3ZEgUkFJMqx6ptqPlcjm3ygeGBAEpxbQ6ltp09Pz8XK8kQ4KAlGJa\nHUntOqpYkiFBQEoxrQ6lNudHhnR5CYW5ib2DAdLqQGo7z2BIl5dQmFvYOyAArI6nELpSu/k6\nz5EuLqEwgL3jGKQUwKonj1NSztpdWkJh6u8dPSCl6lv1HbB1pKo/gbTFkCAgpepbXQqJ0pEh\nUUBK1be6EBKmI0OigJQCWJ09R+J0ZEgUkFIEqzOzdqCODIkCUopp9S5F6siQKCClmFZbqY8n\nkGpOe79DDqn2+wAImd1bTtw/Pj7e1zSZFkekGEgpplUr1bmgoeYlDVvII1IZkWO4ewcOpFVX\nypCuX0JhbmDvoIC0aqQ+JhoM6folFAa7d/BAWr12J+zqXqz6jiFBQEoxrV4PJr6dtbt2CYWB\n7h1EkFavqCeQthgSBKQU0erPH2JHhkQBKUWz+tNWBJPaYEgQkFIkqz+7oahHqv5pkiFBQEpR\nrLYRbXM5lgJM3BkSBKQUwOrPx0j0nsuRFOGpJEOCgJSqbPWnM7Gwy8WQhi6hMMh9FilV0aqJ\naJdGe0zX5nL3ZWlIg5dQmEp7x/kTYkP6oB2J9s57Njebj3fLu+el50hDl1CYOnvHhY3NDmm2\nWbHdKdHeKPN+c9l29Px8f/xrztpdtYTCVNlnLx1+oEOa5xG/c/J0suUAAAd3SURBVErUE9J6\n+aXpqC+k+hjSTNxySDOcg/w5vGChL6TtLUMauoTCGNLVzBLSUUS7P7o8vNneQq4qQ5qLGz5H\nmi6kP/0Rbf7qcnl8s7mFXFWGNBs3PGs3yTnSmYbOg1xVhgQBKTXZrN3giBqQq8qQICClJrEa\nFVEDclVlCynfMfYoCludOyW6HuSqShZSwlmfURS0KtJQC3JV5Qop4/MQoyhkVS6iBuSqQoW0\neqNvCcUwpCDjrcoczXVAripSSKvdh4MlFMOQgoyzKt9QC3JV5QrJc6Qgw60miqgBuapwIfUt\noSDO2oUYZjVhRA3IVcUK6f0c6WtL7fcBmI77+8/69gntKVFtiU/FkJBW66kP7VqqP6L1XXFT\nXarD+5UMr/tfXGLigWgHa1VtYY1I6xQh9V4DWluqw6701/WVF9rNFVEDalW9Y0jzgw/pQ/D1\nmku/54yogbSqdhjS/HyikCZ4lugypFW1w5AqQD9Hui6kGg21kFbVDlJI01/ZsKH+hug5fX8l\nvILHOxfPkapF1FB/+/WACunUEgoD3RCA15TasW16eb9cHwVeNaIG6PabCEOK8Up4lcMDls+P\nz0cN1X9TFeb2m2rBhnQ9f85R74DvLe3Hj7QRDbXgtl+DIVVhU8d7K809p0ak5dnKeikmuQuJ\nE1EDYPsdY0gV6NvjT5wjDTngK5XeJiRWRA3Vt18fhjQju6O1njq2s3aHB3LL57u7u8nPnOYY\n38phSEOXUJi5N8TRPnkipHXPVPPy7i2tu2pnSfeoyY8thjR0CYWZa0OcfEzvDakdj46+UTek\n2EWrc2FIQ5dQmKk3xOWDop7Tofvmrp6QZjm0O0nfE7LVuzKkoUsozET/5shZxdHOuHxsEvrS\nE1LVZ5d6LhGq/7yxIQ1dQmGmeLG2sWflm5CWx3tp3f32OCTA88aGNHQJhSn4GlPFprW2IfUc\nN1U9kjKkazGkMNPMC9/X3j17OT5HMqR+DGnD5cf9iZ9Yea1+Dt9Hz6yd50i9GFLLmb1jpicm\nkXvHwMeciUGuKkNq6DtemfmZfeTewbTKJXWrIdW6NAa5dzCtckndWki1Ly1D7h1Mq1xSsJD2\nD+w7/+b3gDyD7gNplUuKFVJnMqH9Nx+MQJ5B94K0yiWFCqk7mYD83wGYvaPzkPJ6fFd9MKtq\nn4wh5doQQbpPBLwe31UfyqrqQA6p3AuR/99ffz0+Pn7WF64vyf3j4+Ga6rlL5oM0Ii2f75Z3\nB+dINCBSB8+ovR7fVR/IqupCHpHKiGz2hLu7L7uv+//NlU8EIHuHIQ0kS0hHB/59P1RzZ6Hs\nHZ4jDePThrQ3wFwTUvWHXcze4azdID5rSJ1H0YOHVEO6GqRVLqm6IR3+b7TOQ+qJkO7u6r3I\nCHTvYFrlkkKF1KX/HOmu5qv1QPcOplUuqdsL6e6u8qsjAEFa5ZIinSMdcHLWzpAOQVrlkgLN\n2h1iSFeDtMolVTukM5x4QtZZu2OQVrmkbi6kys+WIPcOplUuqdsLqS5IKaZVLilDioGUYlrl\nkjKkGEgpplUuKUOKgZRiWuWSMqQYSCmmVS4pQ4qBlGJa5ZIypBhIKaZVLilDioGUYlrlkjKk\nGEgpplUuKUOKgZRiWuWSMqQYSCmmVS4pQ4qBlGJa5ZIypBhIKaZVLqkhIa0aepZQmFwbYhRI\nq1xSg0I6sYTC5NoQo0Ba5ZIypBhIKaZVLqkBIXU7MiQASKtcUkNC2p0ifW2p/T4AIvUZOiI5\n2UACaZVLauj0tyGRQFrlkjKkGEgpplUuKQ/tYiClmFa5pAaGtDdzZ0j1QVrlkhp6ZUPfEgqT\na0OMAmmVS8pr7WIgpZhWuaQMKQZSimmVS8qQYiClmFa5pAwpBlKKaZVLypBiIKWYVrmkDCkG\nUopplUvKkGIgpZhWuaQMKQZSimmVS6pASCIyPqTJmGyoGwNSimmVVcqQrgEpxbTKKmVI14CU\nYlpllcKFJHKLGJJIAQxJpACGJFIAQxIpgCGJFAAR0mrL5ovdXTWV1juLrQhDqAVntb/5qm6/\nK//kFGaMkPZvf7xcUf1d5GPfgAi14Kz2E6q5/a5tN0NIq7UhXQJntRdSze23ckQ6+qL+ztHS\nOVoBWuGE1ut6IXXXR0fj4Ej48O4SYENiHPx3PvH2W+JqAoW0Ohgk9z+XHs8ZIR1NNnw8cFSl\ncxYN8NnwYUVcTbVCOlwVq21CBwHt3VXUjxHS0ReQ47vVma/qsereqq9F2H5Hf3E/pG3nu8/v\ng2dBQ0M6x7EYAUPqc+gcPKw7x3WHhxSr3bREgnOkw7ursDpxuy6GdI3IuZDWx/eNxZDOcTgL\nxMCQrhf5iCbnZAPlKfuPWyuGUcN+3gipvv23/pUNnbmrHNPfIreOIYkUwJDkU7E6mr6bB0MS\nKYAhiRTAkEQKYEgiBTAkkQIYkkgBDEmkAIYkUgBDEimAIYkUwJBECmBIIgUwJJECGJJIAQxJ\npACGJFIAQxIpgCGJFMCQRApgSCIFMCSRAhiSSAEMSaQAhiRSAEMSKYAhiRTAkEQKYEgiBTAk\nkQIYkkgBDEmkAP8Pb87L9AlErtsAAAAASUVORK5CYII=",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "boreLim <- ggplot(boreDat, aes(x = devStage, y = gExp)) + \n",
    "             geom_jitter(width = 0.2, alpha = 0.5) +\n",
    "             labs(title = \"theBore\") +\n",
    "             theme_bw() +\n",
    "             theme(legend.position = \"none\") +\n",
    "             ylim(5, 10) +\n",
    "             xlab(\"\") +\n",
    "             stat_summary(aes(group=1), fun.y=mean, geom=\"line\", colour=\"red\")\n",
    "\n",
    "hitLim <- ggplot(hitDat, aes(x = devStage, y = gExp)) + \n",
    "             geom_jitter(width = 0.2, alpha = 0.5) +\n",
    "             labs(title = \"theHit\") +\n",
    "             theme_bw() +\n",
    "             theme(legend.position = \"none\") +\n",
    "             ylim(5, 10) +\n",
    "             ylab(\"\") +\n",
    "             xlab(\"\") +\n",
    "             stat_summary(aes(group=1), fun.y=mean, geom=\"line\", colour=\"red\")\n",
    "\n",
    "grid.arrange(boreLim, hitLim, nrow = 2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Slide 24"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A grouped_df: 10 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>devStage</th><th scope=col>gene</th><th scope=col>mean(gExp)</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>E16    </td><td>theBore</td><td>7.544143</td></tr>\n",
       "\t<tr><td>E16    </td><td>theHit </td><td>5.540857</td></tr>\n",
       "\t<tr><td>P2     </td><td>theBore</td><td>7.583500</td></tr>\n",
       "\t<tr><td>P2     </td><td>theHit </td><td>5.844875</td></tr>\n",
       "\t<tr><td>P6     </td><td>theBore</td><td>7.554000</td></tr>\n",
       "\t<tr><td>P6     </td><td>theHit </td><td>5.784250</td></tr>\n",
       "\t<tr><td>P10    </td><td>theBore</td><td>7.571000</td></tr>\n",
       "\t<tr><td>P10    </td><td>theHit </td><td>6.375125</td></tr>\n",
       "\t<tr><td>4_weeks</td><td>theBore</td><td>7.559000</td></tr>\n",
       "\t<tr><td>4_weeks</td><td>theHit </td><td>9.173375</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped_df: 10 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       " devStage & gene & mean(gExp)\\\\\n",
       " <fct> & <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t E16     & theBore & 7.544143\\\\\n",
       "\t E16     & theHit  & 5.540857\\\\\n",
       "\t P2      & theBore & 7.583500\\\\\n",
       "\t P2      & theHit  & 5.844875\\\\\n",
       "\t P6      & theBore & 7.554000\\\\\n",
       "\t P6      & theHit  & 5.784250\\\\\n",
       "\t P10     & theBore & 7.571000\\\\\n",
       "\t P10     & theHit  & 6.375125\\\\\n",
       "\t 4\\_weeks & theBore & 7.559000\\\\\n",
       "\t 4\\_weeks & theHit  & 9.173375\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 10 × 3\n",
       "\n",
       "| devStage &lt;fct&gt; | gene &lt;fct&gt; | mean(gExp) &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| E16     | theBore | 7.544143 |\n",
       "| E16     | theHit  | 5.540857 |\n",
       "| P2      | theBore | 7.583500 |\n",
       "| P2      | theHit  | 5.844875 |\n",
       "| P6      | theBore | 7.554000 |\n",
       "| P6      | theHit  | 5.784250 |\n",
       "| P10     | theBore | 7.571000 |\n",
       "| P10     | theHit  | 6.375125 |\n",
       "| 4_weeks | theBore | 7.559000 |\n",
       "| 4_weeks | theHit  | 9.173375 |\n",
       "\n"
      ],
      "text/plain": [
       "   devStage gene    mean(gExp)\n",
       "1  E16      theBore 7.544143  \n",
       "2  E16      theHit  5.540857  \n",
       "3  P2       theBore 7.583500  \n",
       "4  P2       theHit  5.844875  \n",
       "5  P6       theBore 7.554000  \n",
       "6  P6       theHit  5.784250  \n",
       "7  P10      theBore 7.571000  \n",
       "8  P10      theHit  6.375125  \n",
       "9  4_weeks  theBore 7.559000  \n",
       "10 4_weeks  theHit  9.173375  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "miniDat %>%\n",
    "    group_by(devStage, gene) %>% \n",
    "    summarize(mean(gExp))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "lm(formula = gExp ~ devStage, data = miniDat, subset = gene == \n",
       "    \"theHit\")\n",
       "\n",
       "Residuals:\n",
       "     Min       1Q   Median       3Q      Max \n",
       "-0.78538 -0.13299 -0.04788  0.17062  0.51814 \n",
       "\n",
       "Coefficients:\n",
       "                Estimate Std. Error t value Pr(>|t|)    \n",
       "(Intercept)       5.5409     0.1021  54.249  < 2e-16 ***\n",
       "devStageP2        0.3040     0.1399   2.174   0.0368 *  \n",
       "devStageP6        0.2434     0.1399   1.740   0.0909 .  \n",
       "devStageP10       0.8343     0.1399   5.965 9.56e-07 ***\n",
       "devStage4_weeks   3.6325     0.1399  25.973  < 2e-16 ***\n",
       "---\n",
       "Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1\n",
       "\n",
       "Residual standard error: 0.2702 on 34 degrees of freedom\n",
       "Multiple R-squared:  0.9663,\tAdjusted R-squared:  0.9623 \n",
       "F-statistic: 243.4 on 4 and 34 DF,  p-value: < 2.2e-16\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "summary(lm(gExp ~ devStage, miniDat, subset = gene == \"theHit\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Slide 38"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "twoFactFit <- lm(formula = gExp ~ gType * devStage, data = miniDat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in cbind(sampleMeans = theAvgs, minuRef = theAvgs - theAvgs[\"wt.E16\"], :\n",
      "\"number of rows of result is not a multiple of vector length (arg 3)\""
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 2 × 5 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Irs4</th><th scope=col>Nrl</th><th scope=col>Irs4</th><th scope=col>Nrl</th><th scope=col>twoFactFit</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>wt</th><td>7.765750</td><td>11.244200</td><td>NA</td><td>NA</td><td>6.53737500</td></tr>\n",
       "\t<tr><th scope=row>NrlKO</th><td>7.739684</td><td> 6.089632</td><td>NA</td><td>NA</td><td>0.01195833</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 2 × 5 of type dbl\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & Irs4 & Nrl & Irs4 & Nrl & twoFactFit\\\\\n",
       "\\hline\n",
       "\twt & 7.765750 & 11.244200 & NA & NA & 6.53737500\\\\\n",
       "\tNrlKO & 7.739684 &  6.089632 & NA & NA & 0.01195833\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 2 × 5 of type dbl\n",
       "\n",
       "| <!--/--> | Irs4 | Nrl | Irs4 | Nrl | twoFactFit |\n",
       "|---|---|---|---|---|---|\n",
       "| wt | 7.765750 | 11.244200 | NA | NA | 6.53737500 |\n",
       "| NrlKO | 7.739684 |  6.089632 | NA | NA | 0.01195833 |\n",
       "\n"
      ],
      "text/plain": [
       "      Irs4     Nrl       Irs4 Nrl twoFactFit\n",
       "wt    7.765750 11.244200 NA   NA  6.53737500\n",
       "NrlKO 7.739684  6.089632 NA   NA  0.01195833"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cbind(sampleMeans = theAvgs,\n",
    "      minuRef = theAvgs - theAvgs[\"wt.E16\"],\n",
    "      twoFactFit = coef(twoFactFit))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "&lt;NA&gt;"
      ],
      "text/latex": [
       "<NA>"
      ],
      "text/markdown": [
       "&lt;NA&gt;"
      ],
      "text/plain": [
       "[1] NA"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "theAvgs[\"NrlKO.4_weeks\"] -\n",
    "    (theAvgs[\"wt.E16\"] +\n",
    "        (theAvgs[\"NrlKO.E16\"] - theAvgs[\"wt.E16\"]) +\n",
    "             (theAvgs[\"wt.4_weeks\"] - theAvgs[\"wt.E16\"]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "lm(formula = gExp ~ gType * devStage, data = miniDat)\n",
       "\n",
       "Residuals:\n",
       "     Min       1Q   Median       3Q      Max \n",
       "-1.18437 -0.82788 -0.05175  0.80825  1.43067 \n",
       "\n",
       "Coefficients:\n",
       "                           Estimate Std. Error t value Pr(>|t|)    \n",
       "(Intercept)                 6.53737    0.32754  19.959  < 2e-16 ***\n",
       "gTypeNrlKO                  0.01196    0.50032   0.024    0.981    \n",
       "devStageP2                  0.20350    0.46321   0.439    0.662    \n",
       "devStageP6                  0.06163    0.46321   0.133    0.895    \n",
       "devStageP10                 0.45738    0.46321   0.987    0.327    \n",
       "devStage4_weeks             2.02125    0.46321   4.364 4.46e-05 ***\n",
       "gTypeNrlKO:devStageP2      -0.06533    0.68182  -0.096    0.924    \n",
       "gTypeNrlKO:devStageP6       0.12829    0.68182   0.188    0.851    \n",
       "gTypeNrlKO:devStageP10     -0.05533    0.68182  -0.081    0.936    \n",
       "gTypeNrlKO:devStage4_weeks -0.39683    0.68182  -0.582    0.562    \n",
       "---\n",
       "Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1\n",
       "\n",
       "Residual standard error: 0.9264 on 68 degrees of freedom\n",
       "Multiple R-squared:  0.3829,\tAdjusted R-squared:  0.3012 \n",
       "F-statistic: 4.688 on 9 and 68 DF,  p-value: 7.517e-05\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "summary(twoFactFit)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Slide 40"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(predvars, data, env): object 'prMat' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(predvars, data, env): object 'prMat' not found\nTraceback:\n",
      "1. summary(lm(formula = prMat ~ gType * devStage, data = prDes))",
      "2. lm(formula = prMat ~ gType * devStage, data = prDes)",
      "3. eval(mf, parent.frame())",
      "4. eval(mf, parent.frame())",
      "5. stats::model.frame(formula = prMat ~ gType * devStage, data = prDes, \n .     drop.unused.levels = TRUE)",
      "6. model.frame.default(formula = prMat ~ gType * devStage, data = prDes, \n .     drop.unused.levels = TRUE)",
      "7. eval(predvars, data, env)",
      "8. eval(predvars, data, env)"
     ]
    }
   ],
   "source": [
    "summary(lm(formula = prMat ~ gType * devStage, data = prDes))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Slide 41"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(predvars, data, env): object 'prMatSimple' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(predvars, data, env): object 'prMatSimple' not found\nTraceback:\n",
      "1. summary(lm(formula = prMatSimple ~ gType * devStage, data = prDes))",
      "2. lm(formula = prMatSimple ~ gType * devStage, data = prDes)",
      "3. eval(mf, parent.frame())",
      "4. eval(mf, parent.frame())",
      "5. stats::model.frame(formula = prMatSimple ~ gType * devStage, \n .     data = prDes, drop.unused.levels = TRUE)",
      "6. model.frame.default(formula = prMatSimple ~ gType * devStage, \n .     data = prDes, drop.unused.levels = TRUE)",
      "7. eval(predvars, data, env)",
      "8. eval(predvars, data, env)"
     ]
    }
   ],
   "source": [
    "summary(lm(formula = prMatSimple ~ gType * devStage, data = prDes))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
